---
title: "Exp1 Sarka Janusova Bulk Data"
author: "Juraj Michalik"
date: "2023-07-19"
output: html_document
---

Processing of exp1 of Sarka Janusova Bulk data (GTKO vs. WT mice).

# Setup

First step is loading all required packages.

```{r setup, include=FALSE, echo = F, message = F}
library(glue)
library(DESeq2)
library(GenomicFeatures)
library(GenomicAlignments)
library(tximport)
library(magrittr)
library(ggplot2)
library(pheatmap)
library(RColorBrewer)
library(gplots)
library(dplyr)
library(ggrepel)
library(Rsubread)

# annotation libs
library(AnnotationHub)
```

We also load some functions that will be needed later, notably regarding annotation.

```{r cars}
# Get current Mus Musculus database from AnnotationHub if not downloaded already
getEnsMm106 <- function(){
  aHu <- AnnotationHub()
  aHuQuery <- query(aHu, pattern = c("Mus Musculus", "EnsDb", '106'))
  EnsMm106 <- aHuQuery[[1]]
  return(EnsMm106)
}


# adds annotation to object
reannotate <- function(dds.obj, keyslist = row.names(dds.obj)){
  EnsMm106 <- getEnsMm106()
  symbol.vector <- mapIds(EnsMm106,
                          keys = row.names(dds.obj),
                          column = "SYMBOL",
                          keytype = "GENEID",
                          multiVals = "first")
  # rename only if symbol is a thing
  row.names(dds.obj) <- ifelse(is.na(symbol.vector) | symbol.vector == '', row.names(dds.obj), symbol.vector)
  return(dds.obj)    
}

# returns PCA data with more PCs (12 I think)
get.complete.PCA.data <- function(dds.obj, ntop = 500){
	rvar <- rowVars(assay(dds.obj))
	select <- order(rvar, decreasing=TRUE)[seq_len(min(ntop, length(rvar)))]
	pca.dat <- prcomp(t(assay(dds.obj)[select,]))
	return(pca.dat)
}
```

Finally we setup paths to directories and files, plus some statistics.
Bamfiles were generated by STAR 2.7.10, reference with the same software from the gtf file below and GRCm39 assembly.

```{r}
bam.path <- "Bamfiles1/"
gtf.path <- "Mus_musculus.GRCm39.106.gtf"
sample.path <- "SampleSheets/exp1_Sample_Sheet.csv" # MUST have Experiment field to annotate experiment

# filtering statistic
padj.max <- 0.01
lfc.min <- 0.6

col_WT <- '#55a0fbff'
col_GTKO <- '#c000c0ff'

dir.create(file.path('Rds_data'), showWarnings = F)
dir.create(file.path('Figures'), showWarnings = F)
dir.create(file.path('Tables'), showWarnings = F)
```

# Global Analysis

## Building count table.

First we need to build count table with annotations.

```{r}
files <- list.files(bam.path, pattern = "\\.bam$")
files <- paste0(bam.path, files)

# get reference and set it up
(txdb <- makeTxDbFromGFF(gtf.path, format = "gtf", circ_seqs = character()))
ebg <- exonsBy(txdb, by = "gene")

# quantification step
expcounts <- summarizeOverlaps(features = ebg,
                               reads = files,
                               mode = "Union",
                               ignore.strand = FALSE,
                               preprocess.reads = invertStrand)

# remove bam from colnames
colnames(expcounts) <- gsub('Aligned.sortedByCoord.out.bam', '', colnames(expcounts))
```

Now we adjust and add meta data.

```{r}
sample_tab <- read.csv(sample.path)

# reorder table to match the ordering of count table
sample_tab <- sample_tab[order(match(sample_tab$SampleName, colnames(expcounts))),]
rownames(sample_tab) <- sample_tab$SampleName
geneslist <- rownames(expcounts)
colData(expcounts) <- DataFrame(sample_tab)
colData(expcounts)$Activation <- factor(colData(expcounts)$Activation, levels = c('non-act', '1ku1', '4ku1')) # necessary for DESeq2
colData(expcounts)$Knockout <- factor(colData(expcounts)$Knockout, levels = c('WT', 'GTKO')) # necessary for DESeq2
colData(expcounts)$Experiment <- as.factor(colData(expcounts)$Experiment) # necessary fo DESeq2
```

And we show the counts of reads per each sample.

```{r}
round(colSums(assay(expcounts)) / 1e6, 1)
```

## Preparation of DESeq2 object

Here we prepare DESeq object. We want to take into account KO factor but also potential differences between 1-to-1 (1ku1) and 4-to-1 (4ku1) activation. Finally, there may be differences between Experiments. Design must account for all of these variables. We also remove low-count, noisy genes and reannotate data.

```{r}
dds_exp1 <- DESeqDataSet(expcounts, design = ~ Experiment + Activation + Knockout)
rowData(dds_exp1)$EnsemblID <- rownames(expcounts)
dds_exp1 <- dds_exp1[rowSums(counts(dds_exp1)) > 10, ] # remove <= 10 counts = noise
dds_exp1 <- reannotate(dds_exp1)
```

Write out counts.

```{r}
write.csv(assay(expcounts), 'Tables/GTKO_activation_assay_raw_counts.csv')
```


Run DESeq itself.

```{r}
dds_exp1 <- DESeq(dds_exp1)
```

## Regularized log transformation and visualization

We need to perform regularized log transformation. This is used for purpose of visualization only and nothing else. Since this is done for purpose of publication, we use design-aware approach (as QC was done beforehand. At any rate, results are rather similar).

```{r}
rlog_exp1 <- rlog(dds_exp1, blind = FALSE)
```

Now we can make PCA. We plot PCA for 1-2 and 2-3 components. We include blank versions without text for easier manipulation.

```{r}
pca_data_cmp <- get.complete.PCA.data(rlog_exp1)
pca_data <- pca_data_cmp$x %>% as.data.frame %>% cbind(., sample_tab)

# get fraction of variability captured by each PC
pca_Var <- pca_data_cmp$sdev^2/sum(pca_data_cmp$sdev^2)
```

```{r}
pca1 <- ggplot(pca_data, aes(x = PC1, y = PC2, color = Knockout, shape = Activation)) +
  geom_point(size = 5) +
  geom_text_repel(aes(label = rownames(pca_data)), col='black', seed = 120) +
  scale_color_manual(values = c(col_GTKO, col_WT)) +
  ggtitle(paste0('exp1 - Normalized PCA (design-aware, PC1 + PC2)' )) +
  xlab(paste0("PC1: ", round(pca_Var[1]*100), "% variance")) +
  ylab(paste0("PC2: ", round(pca_Var[2]*100), "% variance")) +
  theme_classic() +
  theme(panel.grid.major = element_line(colour = "grey70", size = 0.3),
        panel.grid.minor = element_line(colour = "grey70", size = 0.1))
print(pca1)
```

```{r}
tiff('Figures/exp1_PCA1_PCA2.tiff', units = 'in', width = 7.5, height = 5.5, res = 150)
print(pca1)
dev.off()
```

```{r}
pca2 <- ggplot(pca_data, aes(x = PC2, y = PC3, color = Knockout, shape = Activation)) +
  geom_point(size = 5) +
  geom_text_repel(aes(label = rownames(pca_data)), col='black', seed = 120) +
  scale_color_manual(values = c(col_GTKO, col_WT)) +
  ggtitle(paste0('exp1 - Normalized PCA (design-aware, PC2 + PC3)' )) +
  xlab(paste0("PC2: ", round(pca_Var[2]*100), "% variance")) +
  ylab(paste0("PC3: ", round(pca_Var[3]*100), "% variance")) +
  theme_classic() +
  theme(panel.grid.major = element_line(colour = "grey70", size = 0.3),
        panel.grid.minor = element_line(colour = "grey70", size = 0.1))
print(pca2)
```

```{r}
tiff('Figures/exp1_PCA2_PCA3.tiff', units = 'in', width = 7.5, height = 5.5, res = 150)
print(pca2)
dev.off()
```

Plot blank versions of pca for manual annotation.

```{r}
pca1 <- ggplot(pca_data, aes(x = PC1, y = PC2, color = Knockout, shape = Activation)) +
  geom_point(size = 5) +
  scale_color_manual(values = c(col_GTKO, col_WT)) +
  xlab('') + 
  ylab('') +
  ggtitle('') +
  theme_classic() +
  theme(panel.grid.major = element_line(colour = "grey70", size = 0.3),
        panel.grid.minor = element_line(colour = "grey70", size = 0.1),
        legend.position = "none",
        axis.text.x=element_blank(),
        axis.text.y=element_blank())
print(pca1)
```

```{r}
tiff('Figures/exp1_PCA1_PCA2_blank.tiff', units = 'in', width = 6.4, height = 5.3, res = 150)
print(pca1)
dev.off()
```

```{r}
pca2 <- ggplot(pca_data, aes(x = PC2, y = PC3, color = Knockout, shape = Activation)) +
  geom_point(size = 5) +
  scale_color_manual(values = c(col_GTKO, col_WT)) +
  theme_classic() +
  xlab('') + 
  ylab('') +
  ggtitle('') +
  theme(panel.grid.major = element_line(colour = "grey70", size = 0.3),
        panel.grid.minor = element_line(colour = "grey70", size = 0.1),
        legend.position = "none",
        axis.text.x=element_blank(),
        axis.text.y=element_blank())
print(pca2)
```

```{r}
tiff('Figures/exp1_PCA2_PCA3_blank.tiff', units = 'in', width = 6.4, height = 5.3, res = 150)
print(pca2)
dev.off()
```
## Results

We now extract results for desired contrast. We want results that are lfc-shrunk, which reduces log2 fold change for genes with small total counts (thus with lower statistical power).

```{r}
res_exp1 <- lfcShrink(dds = dds_exp1, contrast = c('Knockout', 'GTKO', 'WT'), type = 'ashr')
```

We now plot volcano plot. 

```{r}
tmp.tab <- res_exp1 %>% as.data.frame %>% 
  dplyr::filter(!is.na(padj) & !is.na(log2FoldChange)) %>%
  dplyr::mutate(log = -log10(padj), 
                log = ifelse(log == Inf, max(log[log != Inf])*1.02, log),
                legend = case_when(abs(log2FoldChange) <= log2(1.5) & padj >= .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) <= log2(1.5) & padj < .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) > log2(1.5) & padj >= .05  ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   log2FoldChange > log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC > log2(1.5)',
                                   log2FoldChange < log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC < -log2(1.5)'))

xlim.a <- max(abs(min(tmp.tab$log2FoldChange) - 0.1), 
              abs(max(tmp.tab$log2FoldChange) + 0.1))
ylim.a <- max(tmp.tab$log) * 1.08 
diff.x.sig <- xlim.a - 1
colPal <- c(col_WT, col_GTKO, 'gray50')
names(colPal) <- c('p-adj < 0.05 and L2FC < -log2(1.5)', 'p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj >= 0.05 or |L2FC| <= log2(1.5)')
tmp.tab$symbols <- ifelse(rownames(tmp.tab) %in% c('Gzmb', 'Ifng', 'Casp1', 'Tnip1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Tnfrsf4'), rownames(tmp.tab), '')
tmp.tab$symbols[tmp.tab$symbols == 'Tnip1'] <- 'Abin1'
volplot1 <- ggplot(tmp.tab, aes(log2FoldChange, log, label= symbols)) + 
        geom_point(aes(colour=legend)) + xlab('Log2 Fold Change') + ylab('-log10(p.adj)') +
        geom_point(data = tmp.tab %>% as.data.frame() %>% dplyr::filter(symbols %in% c('Gzmb', 'Ifng', 'Casp1', 'Abin1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Tnfrsf4')), pch = 21, size = 4, colour = 'red') +
        scale_color_manual(values=colPal) +
        annotate("rect", xmin = -Inf, xmax = -log2(1.5), ymin = ylim.a/1.06, ymax = Inf, fill = col_WT) +
        annotate("rect", xmin = log2(1.5), xmax =Inf, ymin = ylim.a/1.06, ymax = Inf, fill = col_GTKO) +
        annotate("text", x = -log2(1.5) - diff.x.sig/20, y = ylim.a, hjust = 1, label = "Upregulated in WT", 
                 col = 'white', fontface = 2, size = 3) + 
        annotate("text", x = log2(1.5) + diff.x.sig/20, y = ylim.a, hjust = 0, label = "Upregulated in GTKO", 
                 col = 'white', fontface = 2, size = 3) +
        geom_vline(xintercept = -log2(1.5), linetype="dashed", color = col_WT, size = 0.5) +
        geom_vline(xintercept = log2(1.5), linetype="dashed", color = col_GTKO, size = 0.5) +
        geom_hline(yintercept = -log10(0.05), linetype="dashed", color = "black", size = 0.5) +
        xlim(-xlim.a, xlim.a) + ylim(0, ylim.a) + geom_text_repel(ylim = c(0, ylim.a/1.1), seed = 120, max.overlaps = 10000) + 
        ggtitle("Volcano plot of comparison between GTKO and WT mice") + 
        theme_classic() + 
        theme(
          axis.line.x = element_line(colour = "grey50"),
          axis.line.y = element_line(colour = "grey50"),
          plot.title = element_text(face="bold"))

print(volplot1)
```

Write as tiff.

```{r}
tiff('Figures/exp1_Volcano_plot_KO_vs_WT_all.tiff', units = 'in', width = 6.5, height = 6.5, res = 150)
print(volplot1)
dev.off()
```

Make blank version of the above for manual annotation.

```{r}
volplot1_blank <- ggplot(tmp.tab, aes(log2FoldChange, log)) + 
        geom_point(aes(colour=legend), size = 2) + 
        xlab('') +
        ylab('') +
        geom_point(data = tmp.tab %>% as.data.frame() %>% dplyr::filter(symbols %in% c('Gzmb', 'Ifng', 'Casp1', 'Abin1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Tnfrsf4')), pch = 21, size = 5, colour = 'red') +
        scale_color_manual(values=colPal) +
        annotate("rect", xmin = -Inf, xmax = -log2(1.5), ymin = ylim.a/1.06, ymax = Inf, fill = col_WT) +
        annotate("rect", xmin = log2(1.5), xmax =Inf, ymin = ylim.a/1.06, ymax = Inf, fill = col_GTKO) +
        geom_vline(xintercept = -log2(1.5), linetype="dashed", color = col_WT, size = 0.5) +
        geom_vline(xintercept = log2(1.5), linetype="dashed", color = col_GTKO, size = 0.5) +
        geom_hline(yintercept = -log10(0.05), linetype="dashed", color = "black", size = 0.5) +
        xlim(-xlim.a, xlim.a) + ylim(0, ylim.a) + 
        theme_classic() +
        ggtitle('') +
        theme(
          axis.line.x = element_line(colour = "grey50"),
          axis.line.y = element_line(colour = "grey50"),
          axis.text.x=element_blank(),
          axis.text.y=element_blank(),
          plot.title = element_text(face="bold"),
          legend.position='none'
          )

print(volplot1_blank)
```

Write as tiff.

```{r}
tiff('Figures/exp1_Volcano_plot_KO_vs_WT_all_blank.tiff', units = 'in', width = 4.5, height = 6.5, res = 150)
print(volplot1_blank)
dev.off()
```
Prepare and generate count table with normalized counts and fold changes.

```{r}
norm.counts1 <- counts(dds_exp1, normalized=TRUE)
contrast.tab1 <- res_exp1[,c('log2FoldChange', 'pvalue', 'padj')]
colnames(contrast.tab1) <- paste0(colnames(contrast.tab1), '.GTKO.vs.WT') 
norm.counts1 <- cbind(norm.counts1, contrast.tab1)
write.csv(norm.counts1, 'Tables/Table_exp1_counts_contrast_GTKO_vs_WT_all.csv')
```

# Analyzing activated-only data

We now do similar thing for activated data only.

```{r}
dds_exp1_act <- DESeqDataSet(expcounts[,expcounts$Activation != 'non-act'], design = ~ Experiment + Activation + Knockout)
rowData(dds_exp1_act)$EnsemblID <- rownames(expcounts)
dds_exp1_act <- dds_exp1_act[rowSums(counts(dds_exp1_act)) > 10, ] # remove <= 10 counts = noise
dds_exp1_act <- reannotate(dds_exp1_act)
```

Run DESeq on activated data only.

```{r}
dds_exp1_act <- DESeq(dds_exp1_act)
```

## Regularized log transformation and visualization

We need to perform regularized log transformation, this time for activated samples only.

```{r}
rlog_exp1_act <- rlog(dds_exp1_act, blind = FALSE)
```

We can now plot PCA. At first we need to extract it then plot it. We plot design-aware variant (required for publications).

```{r}
pca_data_cmp_act <- get.complete.PCA.data(rlog_exp1_act)
pca_data_act <- pca_data_cmp_act$x %>% as.data.frame %>% cbind(., sample_tab[sample_tab$Activation != 'non-act',])

# get fraction of variability captured by each PC
pca_Var_act <- pca_data_cmp_act$sdev^2/sum(pca_data_cmp_act$sdev^2)
```

```{r}
pca1_act <- ggplot(pca_data_act, aes(x = PC1, y = PC2, color = Knockout, shape = Activation)) +
  geom_point(size = 5) +
  geom_text_repel(aes(label = rownames(pca_data_act)), col='black', seed = 120) +
  scale_color_manual(values = c(col_GTKO, col_WT)) +
  ggtitle(paste0('exp1 activated - Normalized PCA (design-aware, PC1 + PC2)' )) +
  xlab(paste0("PC1: ", round(pca_Var_act[1]*100), "% variance")) +
  ylab(paste0("PC2: ", round(pca_Var_act[2]*100), "% variance")) +
  theme_classic() +
  theme(panel.grid.major = element_line(colour = "grey70", size = 0.3),
        panel.grid.minor = element_line(colour = "grey70", size = 0.1))
print(pca1_act)
```

```{r}
tiff('Figures/exp1_PCA1_PCA2_act.tiff', units = 'in', width = 7.5, height = 5.5, res = 150)
print(pca1_act)
dev.off()
```

```{r}
pca2_act <- ggplot(pca_data_act, aes(x = PC2, y = PC3, color = Knockout, shape = Activation)) +
  geom_point(size = 5) +
  geom_text_repel(aes(label = rownames(pca_data_act)), col='black', seed = 120) +
  scale_color_manual(values = c(col_GTKO, col_WT)) +
  ggtitle(paste0('exp1 activated - Normalized PCA (design-aware, PC2 + PC3)' )) +
  xlab(paste0("PC2: ", round(pca_Var_act[2]*100), "% variance")) +
  ylab(paste0("PC3: ", round(pca_Var_act[3]*100), "% variance")) +
  theme_classic() +
  theme(panel.grid.major = element_line(colour = "grey70", size = 0.3),
        panel.grid.minor = element_line(colour = "grey70", size = 0.1))
print(pca2_act)
```

```{r}
tiff('Figures/exp1_PCA2_PCA3_act.tiff', units = 'in', width = 7.5, height = 5.5, res = 150)
print(pca2_act)
dev.off()
```

Plot blank versions of pca for manual annotation.

```{r}
pca1_act_blank <- ggplot(pca_data_act, aes(x = PC1, y = PC2, color = Knockout, shape = Activation)) +
  geom_point(size = 5) +
  scale_color_manual(values = c(col_GTKO, col_WT)) +
  xlab('') + 
  ylab('') +
  ggtitle('') +
  theme_classic() +
  theme(panel.grid.major = element_line(colour = "grey70", size = 0.3),
        panel.grid.minor = element_line(colour = "grey70", size = 0.1),
        legend.position = "none",
        axis.text.x=element_blank(),
        axis.text.y=element_blank())
print(pca1_act_blank)
```

```{r}
tiff('Figures/exp1_PCA1_PCA2_act_blank.tiff', units = 'in', width = 6.4, height = 5.3, res = 150)
print(pca1_act_blank)
dev.off()
```

```{r}
pca2_act_blank <- ggplot(pca_data_act, aes(x = PC2, y = PC3, color = Knockout, shape = Activation)) +
  geom_point(size = 5) +
  scale_color_manual(values = c(col_GTKO, col_WT)) +
  theme_classic() +
  xlab('') + 
  ylab('') +
  ggtitle('') +
  theme(panel.grid.major = element_line(colour = "grey70", size = 0.3),
        panel.grid.minor = element_line(colour = "grey70", size = 0.1),
        legend.position = "none",
        axis.text.x=element_blank(),
        axis.text.y=element_blank())
print(pca2_act_blank)
```

```{r}
tiff('Figures/exp1_PCA2_PCA3_act_blank.tiff', units = 'in', width = 6.4, height = 5.3, res = 150)
print(pca2_act_blank)
dev.off()
```

## Results

We now extract results for desired contrast and again we want results that are lfc-shrunk.

```{r}
res_exp1_act <- lfcShrink(dds = dds_exp1_act, contrast = c('Knockout', 'GTKO', 'WT'), type = 'ashr')
```

We now plot volcano plot. 

```{r}
tmp.tab <- res_exp1_act %>% as.data.frame %>% 
  dplyr::filter(!is.na(padj) & !is.na(log2FoldChange)) %>%
  dplyr::mutate(log = -log10(padj), 
                log = ifelse(log == Inf, max(log[log != Inf])*1.02, log),
                legend = case_when(abs(log2FoldChange) <= log2(1.5) & padj >= .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) <= log2(1.5) & padj < .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) > log2(1.5) & padj >= .05  ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   log2FoldChange > log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC > log2(1.5)',
                                   log2FoldChange < log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC < -log2(1.5)'))

xlim.a <- max(abs(min(tmp.tab$log2FoldChange) - 0.1), 
              abs(max(tmp.tab$log2FoldChange) + 0.1))
ylim.a <- max(tmp.tab$log) * 1.08 
diff.x.sig <- xlim.a - 1
colPal <- c(col_WT, col_GTKO, 'gray50')
names(colPal) <- c('p-adj < 0.05 and L2FC < -log2(1.5)', 'p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj >= 0.05 or |L2FC| <= log2(1.5)')
tmp.tab$symbols <- ifelse(rownames(tmp.tab) %in% c('Gzmb', 'Ifng', 'Casp1', 'Tnip1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17f', 'Il17a', 'Rorc'), rownames(tmp.tab), '')
tmp.tab$symbols[tmp.tab$symbols == 'Tnip1'] <- 'Abin1'
volplot1_act <- ggplot(tmp.tab, aes(log2FoldChange, log, label= symbols)) + 
        geom_point(aes(colour=legend)) + xlab('Log2 Fold Change') + ylab('-log10(p.adj)') +
        geom_point(data = tmp.tab %>% as.data.frame() %>% dplyr::filter(symbols %in% c('Gzmb', 'Ifng', 'Casp1', 'Abin1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc')), pch = 21, size = 4, colour = 'red') +
        scale_color_manual(values=colPal) +
        annotate("rect", xmin = -Inf, xmax = -log2(1.5), ymin = ylim.a/1.06, ymax = Inf, fill = col_WT) +
        annotate("rect", xmin = log2(1.5), xmax =Inf, ymin = ylim.a/1.06, ymax = Inf, fill = col_GTKO) +
        annotate("text", x = -log2(1.5) - diff.x.sig/20, y = ylim.a, hjust = 1, label = "Upregulated in WT", 
                 col = 'white', fontface = 2, size = 3) + 
        annotate("text", x = log2(1.5) + diff.x.sig/20, y = ylim.a, hjust = 0, label = "Upregulated in GTKO", 
                 col = 'white', fontface = 2, size = 3) +
        geom_vline(xintercept = -log2(1.5), linetype="dashed", color = col_WT, size = 0.5) +
        geom_vline(xintercept = log2(1.5), linetype="dashed", color = col_GTKO, size = 0.5) +
        geom_hline(yintercept = -log10(0.05), linetype="dashed", color = "black", size = 0.5) +
        xlim(-xlim.a, xlim.a) + ylim(0, ylim.a) + geom_text_repel(ylim = c(0, ylim.a/1.1), seed = 120, max.overlaps = 100) + 
        ggtitle("Volcano plot of comparison between GTKO and WT mice") + 
        theme_classic() + 
        theme(
          axis.line.x = element_line(colour = "grey50"),
          axis.line.y = element_line(colour = "grey50"),
          plot.title = element_text(face="bold"))

print(volplot1_act)
```

Write as tiff.

```{r}
tiff('Figures/exp1_Volcano_plot_KO_vs_WT_act.tiff', units = 'in', width = 6.5, height = 6.5, res = 150)
print(volplot1_act)
dev.off()
```

Make blank version of the above for manual annotation.

```{r}
volplot1_act_blank <- ggplot(tmp.tab, aes(log2FoldChange, log)) + 
        geom_point(aes(colour=legend), size = 2) + 
        xlab('') +
        ylab('') +
        geom_point(data = tmp.tab %>% as.data.frame() %>% dplyr::filter(symbols %in% c('Gzmb', 'Ifng', 'Casp1', 'Abin1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17f', 'Il17a', 'Rorc')), pch = 21, size = 5, colour = 'red') +
        scale_color_manual(values=colPal) +
        annotate("rect", xmin = -Inf, xmax = -log2(1.5), ymin = ylim.a/1.06, ymax = Inf, fill = col_WT) +
        annotate("rect", xmin = log2(1.5), xmax =Inf, ymin = ylim.a/1.06, ymax = Inf, fill = col_GTKO) +
        geom_vline(xintercept = -log2(1.5), linetype="dashed", color = col_WT, size = 0.5) +
        geom_vline(xintercept = log2(1.5), linetype="dashed", color = col_GTKO, size = 0.5) +
        geom_hline(yintercept = -log10(0.05), linetype="dashed", color = "black", size = 0.5) +
        xlim(-xlim.a, xlim.a) + ylim(0, ylim.a) + 
        theme_classic() +
        ggtitle('') +
        theme(
          axis.line.x = element_line(colour = "grey50"),
          axis.line.y = element_line(colour = "grey50"),
          axis.text.x=element_blank(),
          axis.text.y=element_blank(),
          plot.title = element_text(face="bold"),
          legend.position='none'
          )

print(volplot1_act_blank)
```

Write as tiff.

```{r}
tiff('Figures/exp1_Volcano_plot_KO_vs_WT_act_blank.tiff', units = 'in', width = 4.5, height = 6.5, res = 150)
print(volplot1_act_blank)
dev.off()
```

Make heatmap of desired genes.

```{r}
write.csv(res_exp1_act %>% as.data.frame %>% 
          dplyr::filter(padj < 0.01) %>% dplyr::arrange(padj),'exp1_DE_activated.csv')

vis_genes <- c('Ifng', 'Gzmb', 'Prf1', 'Casp1', 'Casp4', 'Serpina3f', 'Serpina3g', 'Tnfrsf4', 'Eomes', 'Ccl1', 'Ccl9', 'Il17a', 'Il17f', 'Il23r', 'Rorc')
```

Make heatmap of top 20 differentially expressed genes.

```{r}
colfunc <- colorRampPalette(c(col_WT, "gray90", col_GTKO))
vis_genes_tab <- limma::removeBatchEffect(assay(rlog_exp1_act)[vis_genes,], colData(rlog_exp1_act)$Experiment)
vis_genes_tab <- vis_genes_tab[,c(7:12,1:6)]

tiff('Figures/exp1_heatmap.tiff', units = 'in', width = 6.5, height = 7.5, res = 150)

pheatmap(vis_genes_tab, border = 'NA', scale = 'row', color=colfunc(64), cluster_rows = F,
         annotation_col = as.data.frame(colData(dds_exp1)[c('Knockout', 'Activation')]), cluster_cols = F,
         main='Heatmap selected genes expressed\n genes between GTKO and WT mice (act. only)', cex = 1,
         annotation_colors = list(Knockout = c(GTKO = col_GTKO, WT = col_WT)))

dev.off()
```

```{r}
colfunc <- colorRampPalette(c(col_WT, "gray90", col_GTKO))
vis_genes_tab <- limma::removeBatchEffect(assay(rlog_exp1_act)[vis_genes,], colData(rlog_exp1_act)$Experiment)
vis_genes_tab <- vis_genes_tab[,c(7:12,1:6)]

tiff('Figures/exp1_heatmap_blank.tiff', units = 'in', width = 4.5, height = 5.9, res = 150)

pheatmap(vis_genes_tab, border = 'NA', scale = 'row', color=colfunc(64), cluster_rows = F,
         annotation_col = as.data.frame(colData(dds_exp1)[c('Knockout', 'Activation')]), cluster_cols = F,
         main='', cex = 1, annotation_names_col = F, annotation_legend = F, legend = F,
         labels_row = rep('', nrow(vis_genes_tab)),
         labels_col = rep('', ncol(vis_genes_tab)),
         annotation_colors = list(Knockout = c(GTKO = col_GTKO, WT = col_WT)))

dev.off()
```

Prepare and generate count table with normalized counts and fold changes.

```{r}
norm.counts.act1 <- counts(dds_exp1_act, normalized=TRUE)
contrast.tab.act1 <- res_exp1_act[,c('log2FoldChange', 'pvalue', 'padj')]
colnames(contrast.tab.act1) <- paste0(colnames(contrast.tab.act1), '.GTKO.vs.WT') 
norm.counts.act1 <- cbind(norm.counts.act1, contrast.tab.act1)
write.csv(norm.counts.act1, 'Tables/Table_exp1_counts_contrast_GTKO_vs_WT_only_activated.csv')
```

# Analyzing non-activated-only data

Finally, similar analysis will be done for non-activated data only.

```{r}
dds_exp1_non_act <- DESeqDataSet(expcounts[,expcounts$Activation == 'non-act'], design = ~ Experiment + Knockout)
rowData(dds_exp1_non_act)$EnsemblID <- rownames(expcounts)
dds_exp1_non_act <- dds_exp1_non_act[rowSums(counts(dds_exp1_non_act)) > 10, ] # remove <= 10 counts = noise
dds_exp1_non_act <- reannotate(dds_exp1_non_act)
```

Run DESeq on activated data only.

```{r}
dds_exp1_non_act <- DESeq(dds_exp1_non_act)
```

## Regularized log transformation and visualization

We need to perform regularized log transformation, this time for non-activated samples only.

```{r}
rlog_exp1_non_act <- rlog(dds_exp1_non_act, blind = FALSE)
```

We can now plot PCA. At first we need to extract it then plot it. We plot design-aware variant (required for publications).

```{r}
pca_data_cmp_non_act <- get.complete.PCA.data(rlog_exp1_non_act)
pca_data_non_act <- pca_data_cmp_non_act$x %>% as.data.frame %>% cbind(., sample_tab[sample_tab$Activation == 'non-act',])

# get fraction of variability captured by each PC
pca_Var_non_act <- pca_data_cmp_non_act$sdev^2/sum(pca_data_cmp_non_act$sdev^2)
```

```{r}
pca1_non_act <- ggplot(pca_data_non_act, aes(x = PC1, y = PC2, color = Knockout, shape = Activation)) +
  geom_point(size = 5) +
  geom_text_repel(aes(label = rownames(pca_data_non_act)), col='black', seed = 120) +
  scale_color_manual(values = c(col_GTKO, col_WT)) +
  ggtitle(paste0('exp1 non-activated - Normalized PCA (design-aware, PC1 + PC2)' )) +
  xlab(paste0("PC1: ", round(pca_Var_non_act[1]*100), "% variance")) +
  ylab(paste0("PC2: ", round(pca_Var_non_act[2]*100), "% variance")) +
  theme_classic() +
  theme(panel.grid.major = element_line(colour = "grey70", size = 0.3),
        panel.grid.minor = element_line(colour = "grey70", size = 0.1))
print(pca1_non_act)
```

```{r}
tiff('Figures/exp1_PCA1_PCA2_non_act.tiff', units = 'in', width = 7.5, height = 5.5, res = 150)
print(pca1_non_act)
dev.off()
```

```{r}
pca2_non_act <- ggplot(pca_data_non_act, aes(x = PC2, y = PC3, color = Knockout, shape = Activation)) +
  geom_point(size = 5) +
  geom_text_repel(aes(label = rownames(pca_data_non_act)), col='black', seed = 120) +
  scale_color_manual(values = c(col_GTKO, col_WT)) +
  ggtitle(paste0('exp1 non-activated - Normalized PCA (design-aware, PC2 + PC3)' )) +
  xlab(paste0("PC2: ", round(pca_Var_non_act[2]*100), "% variance")) +
  ylab(paste0("PC3: ", round(pca_Var_non_act[3]*100), "% variance")) +
  theme_classic() +
  theme(panel.grid.major = element_line(colour = "grey70", size = 0.3),
        panel.grid.minor = element_line(colour = "grey70", size = 0.1))
print(pca2_non_act)
```

```{r}
tiff('Figures/exp1_PCA2_PCA3_non_act.tiff', units = 'in', width = 7.5, height = 5.5, res = 150)
print(pca2_non_act)
dev.off()
```

Plot blank versions of pca for manual annotation.

```{r}
pca1_non_act_blank <- ggplot(pca_data_non_act, aes(x = PC1, y = PC2, color = Knockout, shape = Activation)) +
  geom_point(size = 5) +
  scale_color_manual(values = c(col_GTKO, col_WT)) +
  xlab('') + 
  ylab('') +
  ggtitle('') +
  theme_classic() +
  theme(panel.grid.major = element_line(colour = "grey70", size = 0.3),
        panel.grid.minor = element_line(colour = "grey70", size = 0.1),
        legend.position = "none",
        axis.text.x=element_blank(),
        axis.text.y=element_blank())
print(pca1_non_act_blank)
```

```{r}
tiff('Figures/exp1_PCA1_PCA2_non_act_blank.tiff', units = 'in', width = 6.4, height = 5.3, res = 150)
print(pca1_non_act_blank)
dev.off()
```

```{r}
pca2_non_act_blank <- ggplot(pca_data_non_act, aes(x = PC2, y = PC3, color = Knockout, shape = Activation)) +
  geom_point(size = 5) +
  scale_color_manual(values = c(col_GTKO, col_WT)) +
  theme_classic() +
  xlab('') + 
  ylab('') +
  ggtitle('') +
  theme(panel.grid.major = element_line(colour = "grey70", size = 0.3),
        panel.grid.minor = element_line(colour = "grey70", size = 0.1),
        legend.position = "none",
        axis.text.x=element_blank(),
        axis.text.y=element_blank())
print(pca2_non_act_blank)
```

```{r}
tiff('Figures/exp1_PCA2_PCA3_non_act_blank.tiff', units = 'in', width = 6.4, height = 5.3, res = 150)
print(pca2_non_act_blank)
dev.off()
```

## Results

We now extract results for desired contrast and again we want results that are lfc-shrunk.

```{r}
res_exp1_non_act <- lfcShrink(dds = dds_exp1_non_act, contrast = c('Knockout', 'GTKO', 'WT'), type = 'ashr')
```

We now plot volcano plot. 

```{r}
tmp.tab <- res_exp1_non_act %>% as.data.frame %>% 
  dplyr::filter(!is.na(padj) & !is.na(log2FoldChange)) %>%
  dplyr::mutate(log = -log10(padj), 
                log = ifelse(log == Inf, max(log[log != Inf])*1.02, log),
                legend = case_when(abs(log2FoldChange) <= log2(1.5) & padj >= .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) <= log2(1.5) & padj < .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) > log2(1.5) & padj >= .05  ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   log2FoldChange > log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC > log2(1.5)',
                                   log2FoldChange < log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC < -log2(1.5)'))

xlim.a <- max(abs(min(tmp.tab$log2FoldChange) - 0.1), 
              abs(max(tmp.tab$log2FoldChange) + 0.1))
ylim.a <- max(tmp.tab$log) * 1.08 
diff.x.sig <- xlim.a - 1
colPal <- c(col_WT, col_GTKO, 'gray50')
names(colPal) <- c('p-adj < 0.05 and L2FC < -log2(1.5)', 'p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj >= 0.05 or |L2FC| <= log2(1.5)')
tmp.tab$symbols <- ifelse(rownames(tmp.tab) %in% c('Gzmb', 'Ifng', 'Casp1', 'Tnip1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Tnfrsf4'), rownames(tmp.tab), '')
tmp.tab$symbols[tmp.tab$symbols == 'Tnip1'] <- 'Abin1'
volplot1_non_act <- ggplot(tmp.tab, aes(log2FoldChange, log, label= symbols)) + 
        geom_point(aes(colour=legend)) + xlab('Log2 Fold Change') + ylab('-log10(p.adj)') +
        geom_point(data = tmp.tab %>% as.data.frame() %>% dplyr::filter(symbols %in% c('Gzmb', 'Ifng', 'Casp1', 'Abin1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Tnfrsf4')), pch = 21, size = 4, colour = 'red') +
        scale_color_manual(values=colPal) +
        annotate("rect", xmin = -Inf, xmax = -log2(1.5), ymin = ylim.a/1.06, ymax = Inf, fill = col_WT) +
        annotate("rect", xmin = log2(1.5), xmax =Inf, ymin = ylim.a/1.06, ymax = Inf, fill = col_GTKO) +
        annotate("text", x = -log2(1.5) - diff.x.sig/20, y = ylim.a, hjust = 1, label = "Upregulated in WT", 
                 col = 'white', fontface = 2, size = 3) + 
        annotate("text", x = log2(1.5) + diff.x.sig/20, y = ylim.a, hjust = 0, label = "Upregulated in GTKO", 
                 col = 'white', fontface = 2, size = 3) +
        geom_vline(xintercept = -log2(1.5), linetype="dashed", color = col_WT, size = 0.5) +
        geom_vline(xintercept = log2(1.5), linetype="dashed", color = col_GTKO, size = 0.5) +
        geom_hline(yintercept = -log10(0.05), linetype="dashed", color = "black", size = 0.5) +
        xlim(-xlim.a, xlim.a) + ylim(0, ylim.a) + geom_text_repel(ylim = c(0, ylim.a/1.1), seed = 120, max.overlaps = 10000) + 
        ggtitle("Volcano plot of comparison between GTKO and WT mice") + 
        theme_classic() + 
        theme(
          axis.line.x = element_line(colour = "grey50"),
          axis.line.y = element_line(colour = "grey50"),
          plot.title = element_text(face="bold"))

print(volplot1_non_act)
```

Write as tiff.

```{r}
tiff('Figures/exp1_Volcano_plot_KO_vs_WT_non_act.tiff', units = 'in', width = 6.5, height = 6.5, res = 150)
print(volplot1_non_act)
dev.off()
```

Make blank version of the above for manual annotation.

```{r}
volplot1_non_act_blank <- ggplot(tmp.tab, aes(log2FoldChange, log)) + 
        geom_point(aes(colour=legend), size = 2) + 
        xlab('') +
        ylab('') +
        geom_point(data = tmp.tab %>% as.data.frame() %>% dplyr::filter(symbols %in% c('Gzmb', 'Ifng', 'Casp1', 'Abin1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Tnfrsf4')), pch = 21, size = 5, colour = 'red') +
        scale_color_manual(values=colPal) +
        annotate("rect", xmin = -Inf, xmax = -log2(1.5), ymin = ylim.a/1.06, ymax = Inf, fill = col_WT) +
        annotate("rect", xmin = log2(1.5), xmax =Inf, ymin = ylim.a/1.06, ymax = Inf, fill = col_GTKO) +
        geom_vline(xintercept = -log2(1.5), linetype="dashed", color = col_WT, size = 0.5) +
        geom_vline(xintercept = log2(1.5), linetype="dashed", color = col_GTKO, size = 0.5) +
        geom_hline(yintercept = -log10(0.05), linetype="dashed", color = "black", size = 0.5) +
        xlim(-xlim.a, xlim.a) + ylim(0, ylim.a) + 
        theme_classic() +
        ggtitle('') +
        theme(
          axis.line.x = element_line(colour = "grey50"),
          axis.line.y = element_line(colour = "grey50"),
          axis.text.x=element_blank(),
          axis.text.y=element_blank(),
          plot.title = element_text(face="bold"),
          legend.position='none'
          )

print(volplot1_non_act_blank)
```

Write as tiff.

```{r}
tiff('Figures/exp1_Volcano_plot_KO_vs_WT_non_act_blank.tiff', units = 'in', width = 4.5, height = 6.5, res = 150)
print(volplot1_non_act_blank)
dev.off()
```

Prepare and generate count table with normalized counts and fold changes.

```{r}
norm.counts.non.act1 <- counts(dds_exp1_non_act, normalized=TRUE)
contrast.tab.non.act1 <- res_exp1_non_act[,c('log2FoldChange', 'pvalue', 'padj')]
colnames(contrast.tab.non.act1) <- paste0(colnames(contrast.tab.act1), '.GTKO.vs.WT') 
norm.counts.non.act1 <- cbind(norm.counts.non.act1, contrast.tab.non.act1)
write.csv(norm.counts.non.act1, 'Tables/Table_exp1_counts_contrast_GTKO_vs_WT_only_non_activated.csv')
```

# Saving

Finally write out some of data as .rds objects for future use.

```{r}
res_exp1 <- res_exp1 %>% as.data.frame()
res_exp1$EnsemblID <- rowData(dds_exp1)$EnsemblID

res_exp1_act <- res_exp1_act %>% as.data.frame()
res_exp1_act$EnsemblID <- rowData(dds_exp1_act)$EnsemblID

res_exp1_non_act <- res_exp1_non_act %>% as.data.frame()
res_exp1_non_act$EnsemblID <- rowData(dds_exp1_non_act)$EnsemblID

saveRDS(rlog_exp1, 'Rds_data/exp1_rlog_all.rds')
saveRDS(res_exp1, 'Rds_data/exp1_res_KO_vs_WT_all.rds')

saveRDS(rlog_exp1_act, 'Rds_data/exp1_rlog_act_only.rds')
saveRDS(res_exp1_act, 'Rds_data/exp1_res_KO_vs_WT_act_only.rds')

saveRDS(rlog_exp1_non_act, 'Rds_data/exp1_rlog_non_act_only.rds')
saveRDS(res_exp1_non_act, 'Rds_data/exp1_res_KO_vs_WT_non_act_only.rds')

saveRDS(dds_exp1, 'Rds_data/exp1_DESeq_all.rds')
saveRDS(dds_exp1_act, 'Rds_data/exp1_DESeq_act_only.rds')
saveRDS(dds_exp1_non_act, 'Rds_data/exp1_DESeq_non_act_only.rds')
```
