---
title: "Exp1 Sarka Janusova Bulk Data"
author: "Juraj Michalik"
date: "2023-07-19"
output: html_document
---

Analysis of second experiment from Bulk data (GTKO vs. WT mice - activated and activated + inhibited).

# Setup

First step is loading all required packages.

```{r setup, include=FALSE, echo = F, message = F}
library(glue)
library(DESeq2)
library(GenomicFeatures)
library(GenomicAlignments)
library(tximport)
library(magrittr)
library(ggplot2)
library(pheatmap)
library(RColorBrewer)
library(gplots)
library(dplyr)
library(ggrepel)
library(Rsubread)

# annotation libs
library(AnnotationHub)
```

We also load some functions that will be needed later, notably regarding annotation.

```{r cars}
# Get current Mus Musculus database from AnnotationHub if not downloaded already
getEnsMm106 <- function(){
  aHu <- AnnotationHub()
  aHuQuery <- query(aHu, pattern = c("Mus Musculus", "EnsDb", '106'))
  EnsMm106 <- aHuQuery[[1]]
  return(EnsMm106)
}


# adds annotation to object
reannotate <- function(dds.obj, keyslist = row.names(dds.obj)){
  EnsMm106 <- getEnsMm106()
  symbol.vector <- mapIds(EnsMm106,
                          keys = row.names(dds.obj),
                          column = "SYMBOL",
                          keytype = "GENEID",
                          multiVals = "first")
  # rename only if symbol is a thing
  row.names(dds.obj) <- ifelse(is.na(symbol.vector) | symbol.vector == '', row.names(dds.obj), symbol.vector)
  return(dds.obj)    
}

# returns PCA data with more PCs (12 I think)
get.complete.PCA.data <- function(dds.obj, ntop = 500){
	rvar <- rowVars(assay(dds.obj))
	select <- order(rvar, decreasing=TRUE)[seq_len(min(ntop, length(rvar)))]
	pca.dat <- prcomp(t(assay(dds.obj)[select,]))
	return(pca.dat)
}
```

Finally we setup paths to directories and files, plus some statistics.
Bamfiles were generated by STAR 2.7.10, reference with the same software from the gtf file below and GRCm39 assembly.

```{r}
bam.path <- "Bamfiles2/"
gtf.path <- "Mus_musculus.GRCm39.106.gtf"
sample.path <- "SampleSheets/exp2_Sample_Sheet.csv" # MUST have Experiment field to annotate experiment

# filtering statistic
padj.max <- 0.01
lfc.min <- 0.6

col_WT <- '#55a0fbff'
col_GTKO <- '#c000c0ff'

col_Act_WT <- '#55a0fbff'
col_Inh_WT <- '#37c871ff'

col_Act_GTKO <- '#c000c0ff'
col_Inh_GTKO <- '#ffd42aff'

col_Act <- 'firebrick'
col_Inh <- 'steelblue'

dir.create(file.path('Rds_data'), showWarnings = F)
dir.create(file.path('Figures'), showWarnings = F)
dir.create(file.path('Tables'), showWarnings = F)
```

# Building count table.

First we need to build count table with annotations.

```{r}
files <- list.files(bam.path, pattern = "\\.bam$")
files <- paste0(bam.path, files)

# get reference and set it up
(txdb <- makeTxDbFromGFF(gtf.path, format = "gtf", circ_seqs = character()))
ebg <- exonsBy(txdb, by = "gene")

# quantification step
expcounts <- summarizeOverlaps(features = ebg,
                               reads = files,
                               mode = "Union",
                               ignore.strand = FALSE,
                               preprocess.reads = invertStrand)

# remove bam from colnames
colnames(expcounts) <- gsub('Aligned.sortedByCoord.out.bam', '', colnames(expcounts))
```

Now we adjust and add meta data.

```{r}
sample_tab <- read.csv(sample.path)

# reorder table to match the ordering of count table
sample_tab <- sample_tab[order(match(sample_tab$SampleName, colnames(expcounts))),]
rownames(sample_tab) <- sample_tab$SampleName
geneslist <- rownames(expcounts)
colData(expcounts) <- DataFrame(sample_tab)
colData(expcounts)$Activation <- factor(colData(expcounts)$Activation, levels = c('Act', 'Act_and_p38i')) # necessary for DESeq2
colData(expcounts)$Knockout <- factor(colData(expcounts)$Knockout, levels = c('WT', 'GTKO')) # necessary for DESeq2
colData(expcounts)$Experiment <- as.factor(colData(expcounts)$Experiment) # necessary for DESeq2

# rename samples to something a tad bit shorter
colnames(expcounts) <- gsub('1ku1', '1:1', colnames(expcounts))
colnames(expcounts) <- gsub('B26_', '', colnames(expcounts))
colnames(expcounts) <- gsub('^..', '', colnames(expcounts))
colnames(expcounts) <- gsub('activated', 'Act', colnames(expcounts))
colnames(expcounts) <- gsub('plusp38i', '+p38i', colnames(expcounts))
colData(expcounts)$SampleName <- colnames(expcounts)
```

And we show the counts of reads per each sample.

```{r}
round(colSums(assay(expcounts)) / 1e6, 1)
```

# Preparation of DESeq2 object

Here we prepare DESeq object. We want to take into account KO factor but also activation/inhibition. Finally, there may be differences between Experiments. Design must account for all of these variables. We also remove low-count, noisy genes and reannotate data. We don't assume there are interaction factors.

```{r}
dds_exp2 <- DESeqDataSet(expcounts, design = ~ Experiment + Knockout + Activation)
rowData(dds_exp2)$EnsemblID <- rownames(expcounts)
dds_exp2 <- dds_exp2[rowSums(counts(dds_exp2)) > 10, ] # remove <= 10 counts = noise
dds_exp2 <- reannotate(dds_exp2)
```

Write out counts.

```{r}
write.csv(assay(expcounts), 'Tables/p38i_inhibition_assay_raw_counts.csv')
```

Run DESeq itself.

```{r}
dds_exp2 <- DESeq(dds_exp2)
```

# Regularized log transformation and visualization

We need to perform regularized log transformation. This is used for purpose of visualization only and nothing else. Since this is done for purpose of publication, we use design-aware approach (as QC was done beforehand; to reproduce replace blind = FALSE below with blind = TRUE. Here the results are mostly similar.).

```{r}
rlog_exp2 <- rlog(dds_exp2, blind = FALSE)
```

We can now plot PCA. At first we need to extract it then plot it. We plot design-aware variant (required for publications).

```{r}
pca_data_cmp <- get.complete.PCA.data(rlog_exp2)
pca_data <- pca_data_cmp$x %>% as.data.frame %>% cbind(., sample_tab)

# get fraction of variability captured by each PC
pca_Var <- pca_data_cmp$sdev^2/sum(pca_data_cmp$sdev^2)
```

```{r}
pca1 <- ggplot(pca_data, aes(x = PC1, y = PC2, color = Knockout, shape = Activation)) +
  geom_point(size = 5) +
  geom_text_repel(aes(label = rownames(pca_data)), col='black', seed = 120) +
  scale_color_manual(values = c(col_GTKO, col_WT)) +
  ggtitle(paste0('exp2 - Normalized PCA (design-aware, PC1 + PC2)' )) +
  xlab(paste0("PC1: ", round(pca_Var[1]*100), "% variance")) +
  ylab(paste0("PC2: ", round(pca_Var[2]*100), "% variance")) +
  theme_classic() +
  theme(panel.grid.major = element_line(colour = "grey70", size = 0.3),
        panel.grid.minor = element_line(colour = "grey70", size = 0.1))

print(pca1)
```

```{r}
tiff('Figures/exp2_PCA1_PCA2_all.tiff', units = 'in', width = 7.5, height = 5.5, res = 150)
print(pca1)
dev.off()
```

Plot PCA - 2nd and 3rd dimensions.

```{r}
pca2 <- ggplot(pca_data, aes(x = PC2, y = PC3, color = Knockout, shape = Activation)) +
  geom_point(size = 5) +
  geom_text_repel(aes(label = rownames(pca_data)), col='black', seed = 120) +
  scale_color_manual(values = c(col_GTKO, col_WT)) +
  ggtitle(paste0('exp2 - Normalized PCA (design-aware, PC2 + PC3)' )) +
  xlab(paste0("PC2: ", round(pca_Var[2]*100), "% variance")) +
  ylab(paste0("PC3: ", round(pca_Var[3]*100), "% variance")) +
  theme_classic() +
  theme(panel.grid.major = element_line(colour = "grey70", size = 0.3),
        panel.grid.minor = element_line(colour = "grey70", size = 0.1))

print(pca2)
```

```{r}
tiff('Figures/exp2_PCA2_PCA3_all.tiff', units = 'in', width = 7.5, height = 5.5, res = 150)
print(pca2)
dev.off()
```
```{r}
pca1_blank <- ggplot(pca_data, aes(x = PC1, y = PC2, color = Knockout, shape = Activation)) +
  geom_point(size = 5) +
  scale_color_manual(values = c(col_GTKO, col_WT)) +
  xlab('') + 
  ylab('') +
  ggtitle('') +
  theme_classic() +
  theme(panel.grid.major = element_line(colour = "grey70", size = 0.3),
        panel.grid.minor = element_line(colour = "grey70", size = 0.1),
        legend.position = "none",
        axis.text.x=element_blank(),
        axis.text.y=element_blank())

print(pca1_blank)
```

```{r}
tiff('Figures/exp2_PCA1_PCA2_all_blank.tiff', units = 'in', width = 6.1, height = 5.1, res = 150)
print(pca1_blank)
dev.off()
```

```{r}
pca2_blank <- ggplot(pca_data, aes(x = PC2, y = PC3, color = Knockout, shape = Activation)) +
  geom_point(size = 5) +
  scale_color_manual(values = c(col_GTKO, col_WT)) +
  xlab('') + 
  ylab('') +
  ggtitle('') +
  theme_classic() +
  theme(panel.grid.major = element_line(colour = "grey70", size = 0.3),
        panel.grid.minor = element_line(colour = "grey70", size = 0.1),
        legend.position = "none",
        axis.text.x=element_blank(),
        axis.text.y=element_blank())

print(pca2_blank)
```

```{r}
tiff('Figures/exp2_PCA2_PCA3_all_blank.tiff', units = 'in', width = 6.1, height = 5.1, res = 150)
print(pca2_blank)
dev.off()
```

# Results

We now extract results for desired contrast. We want results that are lfc-shrunk, which reduces log2 fold change for genes with small total counts (thus with lower statistical power).

```{r}
res_exp2 <- lfcShrink(dds = dds_exp2, contrast = c('Knockout', 'GTKO', 'WT'), type = 'ashr')
```

We now plot volcano plot. 

```{r}
tmp.tab <- res_exp2 %>% as.data.frame %>% 
  dplyr::filter(!is.na(padj) & !is.na(log2FoldChange)) %>%
  dplyr::mutate(log = -log10(padj), 
                log = ifelse(log == Inf, max(log[log != Inf])*1.02, log),
                legend = case_when(abs(log2FoldChange) <= log2(1.5) & padj >= .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) <= log2(1.5) & padj < .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) > log2(1.5) & padj >= .05  ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   log2FoldChange > log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC > log2(1.5)',
                                   log2FoldChange < log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC < -log2(1.5)'))
  
xlim.a <- max(abs(min(tmp.tab$log2FoldChange) - 0.1), 
              abs(max(tmp.tab$log2FoldChange) + 0.1))
ylim.a <- max(tmp.tab$log) * 1.08 
diff.x.sig <- xlim.a - 1
colPal <- c(col_WT, col_GTKO, 'gray50')
names(colPal) <- c('p-adj < 0.05 and L2FC < -log2(1.5)', 'p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj >= 0.05 or |L2FC| <= log2(1.5)')
tmp.tab$symbols <- ifelse(rownames(tmp.tab) %in% c('Gzmb', 'Ifng', 'Casp1', 'Tnip1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc'), rownames(tmp.tab), '')
tmp.tab$symbols[tmp.tab$symbols == 'Tnip1'] <- 'Abin1'
volplot1 <- ggplot(tmp.tab, aes(log2FoldChange, log, label= symbols)) + 
        geom_point(aes(colour=legend)) + xlab('Log2 Fold Change') + ylab('-log10(p.adj)') +
        geom_point(data = tmp.tab %>% as.data.frame() %>% dplyr::filter(symbols %in% c('Gzmb', 'Ifng', 'Casp1', 'Abin1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc') & legend %in% c('p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj < 0.05 and L2FC < -log2(1.5)')), pch = 21, size = 4, colour = 'red') +
        scale_color_manual(values=colPal) +
        annotate("rect", xmin = -Inf, xmax = -log2(1.5), ymin = ylim.a/1.06, ymax = Inf, fill = col_WT) +
        annotate("rect", xmin = log2(1.5), xmax =Inf, ymin = ylim.a/1.06, ymax = Inf, fill = col_GTKO) +
        annotate("text", x = -log2(1.5) - diff.x.sig/20, y = ylim.a, hjust = 1, label = "Upregulated in WT", 
                 col = 'white', fontface = 2, size = 3) + 
        annotate("text", x = log2(1.5) + diff.x.sig/20, y = ylim.a, hjust = 0, label = "Upregulated in GTKO", 
                 col = 'white', fontface = 2, size = 3) +
        geom_vline(xintercept = -log2(1.5), linetype="dashed", color = col_WT, size = 0.5) +
        geom_vline(xintercept = log2(1.5), linetype="dashed", color = col_GTKO, size = 0.5) +
        geom_hline(yintercept = -log10(0.05), linetype="dashed", color = "black", size = 0.5) +
        xlim(-xlim.a, xlim.a) + ylim(0, ylim.a) + geom_text_repel(ylim = c(0, ylim.a/1.1), seed = 120, max.overlaps = 30000) + 
        ggtitle("Volcano plot of comparison between GTKO and WT mice") + 
        theme_classic() + 
        theme(
          axis.line.x = element_line(colour = "grey50"),
          axis.line.y = element_line(colour = "grey50"),
          plot.title = element_text(face="bold"))

print(volplot1)
```

Write as tiff.

```{r}
tiff('Figures/exp2_Volcano_plot_WT_vs_GTKO_all.tiff', units = 'in', width = 6.5, height = 6.5, res = 150)
print(volplot1)
dev.off()
```

We make blank version.

```{r}
tmp.tab <- res_exp2 %>% as.data.frame %>% 
  dplyr::filter(!is.na(padj) & !is.na(log2FoldChange)) %>%
  dplyr::mutate(log = -log10(padj), 
                log = ifelse(log == Inf, max(log[log != Inf])*1.02, log),
                legend = case_when(abs(log2FoldChange) <= log2(1.5) & padj >= .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) <= log2(1.5) & padj < .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) > log2(1.5) & padj >= .05  ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   log2FoldChange > log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC > log2(1.5)',
                                   log2FoldChange < log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC < -log2(1.5)'))
  
xlim.a <- max(abs(min(tmp.tab$log2FoldChange) - 0.1), 
              abs(max(tmp.tab$log2FoldChange) + 0.1))
ylim.a <- max(tmp.tab$log) * 1.08 
diff.x.sig <- xlim.a - 1
colPal <- c(col_WT, col_GTKO, 'gray50')
names(colPal) <- c('p-adj < 0.05 and L2FC < -log2(1.5)', 'p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj >= 0.05 or |L2FC| <= log2(1.5)')
tmp.tab$symbols <- ifelse(rownames(tmp.tab) %in% c('Gzmb', 'Ifng', 'Casp1', 'Tnip1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc'), rownames(tmp.tab), '')
tmp.tab$symbols[tmp.tab$symbols == 'Tnip1'] <- 'Abin1'
volplot1_blank <- ggplot(tmp.tab, aes(log2FoldChange, log)) + 
        geom_point(aes(colour=legend), size = 2) + 
        xlab('') +
        ylab('') +
        geom_point(data = tmp.tab %>% as.data.frame() %>% dplyr::filter(symbols %in% c('Gzmb', 'Ifng', 'Casp1', 'Abin1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc') & legend %in% c('p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj < 0.05 and L2FC < -log2(1.5)')), pch = 21, size = 5, colour = 'red') +
        scale_color_manual(values=colPal) +
        annotate("rect", xmin = -Inf, xmax = -log2(1.5), ymin = ylim.a/1.06, ymax = Inf, fill = col_WT) +
        annotate("rect", xmin = log2(1.5), xmax =Inf, ymin = ylim.a/1.06, ymax = Inf, fill = col_GTKO) +
        geom_vline(xintercept = -log2(1.5), linetype="dashed", color = col_WT, size = 0.5) +
        geom_vline(xintercept = log2(1.5), linetype="dashed", color = col_GTKO, size = 0.5) +
        geom_hline(yintercept = -log10(0.05), linetype="dashed", color = "black", size = 0.5) +
        xlim(-xlim.a, xlim.a) + ylim(0, ylim.a) + 
        theme_classic() +
        ggtitle('') +
        theme(
          axis.line.x = element_line(colour = "grey50"),
          axis.line.y = element_line(colour = "grey50"),
          axis.text.x=element_blank(),
          axis.text.y=element_blank(),
          plot.title = element_text(face="bold"),
          legend.position='none'
          )

print(volplot1_blank)
```

Write as tiff.

```{r}
tiff('Figures/exp2_Volcano_plot_WT_vs_GTKO_all_blank.tiff', units = 'in', width = 4.5, height = 6.5, res = 150)
print(volplot1_blank)
dev.off()
```

We also extract results where we compare activated and inhibited cells.

```{r}
res_exp2_act_vs_inh <- lfcShrink(dds = dds_exp2, contrast = c('Activation', 'Act', 'Act_and_p38i'), type = 'ashr')
```

Prepare and generate count table with normalized counts and fold changes including fold changes for activation and GTKO vs. WT.

```{r}
norm.counts2 <- counts(dds_exp2, normalized=TRUE)
contrast.tab2 <- res_exp2[,c('log2FoldChange', 'pvalue', 'padj')]
contrast.tab2inh <- res_exp2_act_vs_inh[,c('log2FoldChange', 'pvalue', 'padj')]
colnames(contrast.tab2) <- paste0(colnames(contrast.tab2), '.GTKO.vs.WT') 
colnames(contrast.tab2inh) <- paste0(colnames(contrast.tab2inh), '.Act.vs.Inh') 
norm.counts2 <- cbind(norm.counts2, contrast.tab2, contrast.tab2inh)
write.csv(norm.counts2, 'Tables/Table_exp2_counts_contrast_GTKO_vs_WT_Act_vs_Inh_all.csv')
write.csv(counts(dds_exp2, normalized=TRUE), 'Tables/Table_exp2_norm_counts.csv')
```

# Expression using only activated cells

Now we create object using only activated samples. We split data instead of just using constrast again to stey coherent with first experiment, with which the results of activated subset would be compared; therefore it is necessary to use same approach, even if in this case the separation of activated and inhibited samples smaller according to blinded PCA than in first experiment. However, this may be because sample 17 seems to be significant outlier.

```{r}
dds_exp2_act <- DESeqDataSet(expcounts[,expcounts$Activation == 'Act'], design = ~ Experiment + Knockout)
rowData(dds_exp2_act)$EnsemblID <- rownames(expcounts)
dds_exp2_act <- dds_exp2_act[rowSums(counts(dds_exp2_act)) > 10, ] # remove <= 10 counts = noise
dds_exp2_act <- reannotate(dds_exp2_act)
```

Run DESeq itself.

```{r}
dds_exp2_act <- DESeq(dds_exp2_act)
```

Extract Regularized log-normalized data for activated samples only.

```{r}
rlog_exp2_act <- rlog(dds_exp2_act, blind = FALSE)
```

We can now plot PCA. At first we need to extract it then plot it. We plot design-aware variant (for publications).

```{r}
pca_data_cmp <- get.complete.PCA.data(rlog_exp2_act)
pca_data <- pca_data_cmp$x %>% as.data.frame %>% cbind(., sample_tab[sample_tab$Activation == 'Act',])

# get fraction of variability captured by each PC
pca_Var <- pca_data_cmp$sdev^2/sum(pca_data_cmp$sdev^2)
```

Plot PCA of activated samples only (default).

```{r}
pca1 <- ggplot(pca_data, aes(x = PC1, y = PC2, color = Knockout)) +
  geom_point(size = 3) +
  geom_text_repel(aes(label = rownames(pca_data)), col='black', seed = 120) +
  scale_color_manual(values = c(col_GTKO, col_WT)) +
  ggtitle(paste0('exp2 (activated only) - Normalized PCA (design-aware, PC1 + PC2)' )) +
  xlab(paste0("PC1: ", round(pca_Var[1]*100), "% variance")) +
  ylab(paste0("PC2: ", round(pca_Var[2]*100), "% variance")) +
  theme_classic() +
  theme(panel.grid.major = element_line(colour = "grey70", size = 0.3),
        panel.grid.minor = element_line(colour = "grey70", size = 0.1))

print(pca1)
```

# Getting results for activated data only

We now get differentially expressed genes for cells from activated mice only.

```{r}
res_exp2_act <- lfcShrink(dds = dds_exp2_act, contrast = c('Knockout', 'GTKO', 'WT'), type = 'ashr')
```

We now plot volcano plot for activated cells only.

```{r}
tmp.tab <- res_exp2_act %>% as.data.frame %>% 
  dplyr::filter(!is.na(padj) & !is.na(log2FoldChange)) %>%
  dplyr::mutate(log = -log10(padj), 
                log = ifelse(log == Inf, max(log[log != Inf])*1.02, log),
                legend = case_when(abs(log2FoldChange) <= log2(1.5) & padj >= .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) <= log2(1.5) & padj < .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) > log2(1.5) & padj >= .05  ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   log2FoldChange > log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC > log2(1.5)',
                                   log2FoldChange < log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC < -log2(1.5)'))
  
xlim.a <- max(abs(min(tmp.tab$log2FoldChange) - 0.1), 
              abs(max(tmp.tab$log2FoldChange) + 0.1))
ylim.a <- max(tmp.tab$log) * 1.08 
diff.x.sig <- xlim.a - 1
colPal <- c(col_WT, col_GTKO, 'gray50')
names(colPal) <- c('p-adj < 0.05 and L2FC < -log2(1.5)', 'p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj >= 0.05 or |L2FC| <= log2(1.5)')
tmp.tab$symbols <- ifelse(rownames(tmp.tab) %in% c('Gzmb', 'Ifng', 'Casp1', 'Tnip1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc'), rownames(tmp.tab), '')
tmp.tab$symbols[tmp.tab$symbols == 'Tnip1'] <- 'Abin1'
volplot1_act <- ggplot(tmp.tab, aes(log2FoldChange, log, label= symbols)) + 
        geom_point(aes(colour=legend)) + xlab('Log2 Fold Change') + ylab('-log10(p.adj)') +
        geom_point(data = tmp.tab %>% as.data.frame() %>% dplyr::filter(symbols %in% c('Gzmb', 'Ifng', 'Casp1', 'Abin1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc') & legend %in% c('p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj < 0.05 and L2FC < -log2(1.5)')), pch = 21, size = 4, colour = 'red') +
        scale_color_manual(values=colPal) +
        annotate("rect", xmin = -Inf, xmax = -log2(1.5), ymin = ylim.a/1.06, ymax = Inf, fill = col_WT) +
        annotate("rect", xmin = log2(1.5), xmax =Inf, ymin = ylim.a/1.06, ymax = Inf, fill = col_GTKO) +
        annotate("text", x = -log2(1.5) - diff.x.sig/20, y = ylim.a, hjust = 1, label = "Upregulated in WT", 
                 col = 'white', fontface = 2, size = 3) + 
        annotate("text", x = log2(1.5) + diff.x.sig/20, y = ylim.a, hjust = 0, label = "Upregulated in GTKO", 
                 col = 'white', fontface = 2, size = 3) +
        geom_vline(xintercept = -log2(1.5), linetype="dashed", color = col_WT, size = 0.5) +
        geom_vline(xintercept = log2(1.5), linetype="dashed", color = col_GTKO, size = 0.5) +
        geom_hline(yintercept = -log10(0.05), linetype="dashed", color = "black", size = 0.5) +
        xlim(-xlim.a, xlim.a) + ylim(0, ylim.a) + geom_text_repel(ylim = c(0, ylim.a/1.1), seed = 120, max.overlaps = 30000) + 
        ggtitle("Volcano plot of comparison between GTKO and WT mice \n (activated only)") + 
        theme_classic() + 
        theme(
          axis.line.x = element_line(colour = "grey50"),
          axis.line.y = element_line(colour = "grey50"),
          plot.title = element_text(face="bold"))

print(volplot1_act)
```

Write as tiff.

```{r}
tiff('Figures/exp2_Volcano_plot_WT_vs_GTKO_act.tiff', units = 'in', width = 6.5, height = 6.5, res = 150)
print(volplot1_act)
dev.off()
```

We make blank version.

```{r}
tmp.tab <- res_exp2_act %>% as.data.frame %>% 
  dplyr::filter(!is.na(padj) & !is.na(log2FoldChange)) %>%
  dplyr::mutate(log = -log10(padj), 
                log = ifelse(log == Inf, max(log[log != Inf])*1.02, log),
                legend = case_when(abs(log2FoldChange) <= log2(1.5) & padj >= .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) <= log2(1.5) & padj < .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) > log2(1.5) & padj >= .05  ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   log2FoldChange > log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC > log2(1.5)',
                                   log2FoldChange < log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC < -log2(1.5)'))
  
xlim.a <- max(abs(min(tmp.tab$log2FoldChange) - 0.1), 
              abs(max(tmp.tab$log2FoldChange) + 0.1))
ylim.a <- max(tmp.tab$log) * 1.08 
diff.x.sig <- xlim.a - 1
colPal <- c(col_WT, col_GTKO, 'gray50')
names(colPal) <- c('p-adj < 0.05 and L2FC < -log2(1.5)', 'p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj >= 0.05 or |L2FC| <= log2(1.5)')
tmp.tab$symbols <- ifelse(rownames(tmp.tab) %in% c('Gzmb', 'Ifng', 'Casp1', 'Tnip1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc'), rownames(tmp.tab), '')
tmp.tab$symbols[tmp.tab$symbols == 'Tnip1'] <- 'Abin1'
volplot1_act_blank <- ggplot(tmp.tab, aes(log2FoldChange, log)) + 
        geom_point(aes(colour=legend), size = 2) + 
        xlab('') +
        ylab('') +
        geom_point(data = tmp.tab %>% as.data.frame() %>% dplyr::filter(symbols %in% c('Gzmb', 'Ifng', 'Casp1', 'Abin1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc') & legend %in% c('p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj < 0.05 and L2FC < -log2(1.5)')), pch = 21, size = 5, colour = 'red') +
        scale_color_manual(values=colPal) +
        annotate("rect", xmin = -Inf, xmax = -log2(1.5), ymin = ylim.a/1.06, ymax = Inf, fill = col_WT) +
        annotate("rect", xmin = log2(1.5), xmax =Inf, ymin = ylim.a/1.06, ymax = Inf, fill = col_GTKO) +
        geom_vline(xintercept = -log2(1.5), linetype="dashed", color = col_WT, size = 0.5) +
        geom_vline(xintercept = log2(1.5), linetype="dashed", color = col_GTKO, size = 0.5) +
        geom_hline(yintercept = -log10(0.05), linetype="dashed", color = "black", size = 0.5) +
        xlim(-xlim.a, xlim.a) + ylim(0, ylim.a) + 
        theme_classic() +
        ggtitle('') +
        theme(
          axis.line.x = element_line(colour = "grey50"),
          axis.line.y = element_line(colour = "grey50"),
          axis.text.x=element_blank(),
          axis.text.y=element_blank(),
          plot.title = element_text(face="bold"),
          legend.position='none'
          )

print(volplot1_act_blank)
```

Write as tiff.

```{r}
tiff('Figures/exp2_Volcano_plot_WT_vs_GTKO_act_blank.tiff', units = 'in', width = 4.5, height = 6.5, res = 150)
print(volplot1_act_blank)
dev.off()
```

Prepare and generate count table with normalized counts (GTKO vs. WT).

```{r}
norm.counts.act2 <- counts(dds_exp2_act, normalized=TRUE)
contrast.tab.act2 <- res_exp2_act[,c('log2FoldChange', 'pvalue', 'padj')]
colnames(contrast.tab.act2) <- paste0(colnames(contrast.tab.act2), '.GTKO.vs.WT') 
norm.counts.act2 <- cbind(norm.counts.act2, contrast.tab.act2)
write.csv(norm.counts.act2, 'Tables/Table_exp2_counts_contrast_GTKO_vs_WT_only_activated.csv')
```

# Getting results for inhibited data only

Now we create object using only inhibited samples, the reasoning is the same as above.

```{r}
dds_exp2_inh <- DESeqDataSet(expcounts[,expcounts$Activation != 'Act'], design = ~ Experiment + Knockout)
rowData(dds_exp2_inh)$EnsemblID <- rownames(expcounts)
dds_exp2_inh <- dds_exp2_inh[rowSums(counts(dds_exp2_inh)) > 10, ] # remove <= 10 counts = noise
dds_exp2_inh <- reannotate(dds_exp2_inh)
```

Run DESeq itself.

```{r}
dds_exp2_inh <- DESeq(dds_exp2_inh)
```

Extract regularized log-normalized data.

```{r}
rlog_exp2_inh <- rlog(dds_exp2_inh, blind = FALSE)
```

We now get differentially expressed genes for cells for inhibited cells only.

```{r}
res_exp2_inh <- lfcShrink(dds = dds_exp2_inh, contrast = c('Knockout', 'GTKO', 'WT'), type = 'ashr')
```

We now plot volcano plot for inhibited data only.

```{r}
tmp.tab <- res_exp2_inh %>% as.data.frame %>% 
  dplyr::filter(!is.na(padj) & !is.na(log2FoldChange)) %>%
  dplyr::mutate(log = -log10(padj), 
                log = ifelse(log == Inf, max(log[log != Inf])*1.02, log),
                legend = case_when(abs(log2FoldChange) <= log2(1.5) & padj >= .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) <= log2(1.5) & padj < .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) > log2(1.5) & padj >= .05  ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   log2FoldChange > log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC > log2(1.5)',
                                   log2FoldChange < log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC < -log2(1.5)'))
  
xlim.a <- max(abs(min(tmp.tab$log2FoldChange) - 0.1), 
              abs(max(tmp.tab$log2FoldChange) + 0.1))
ylim.a <- max(tmp.tab$log) * 1.08 
diff.x.sig <- xlim.a - 1
colPal <- c(col_WT, col_GTKO, 'gray50')
names(colPal) <- c('p-adj < 0.05 and L2FC < -log2(1.5)', 'p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj >= 0.05 or |L2FC| <= log2(1.5)')
tmp.tab$symbols <- ifelse(rownames(tmp.tab) %in% c('Gzmb', 'Ifng', 'Casp1', 'Tnip1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc'), rownames(tmp.tab), '')
tmp.tab$symbols[tmp.tab$symbols == 'Tnip1'] <- 'Abin1'
volplot1_inh <- ggplot(tmp.tab, aes(log2FoldChange, log, label= symbols)) + 
        geom_point(aes(colour=legend)) + xlab('Log2 Fold Change') + ylab('-log10(p.adj)') +
        geom_point(data = tmp.tab %>% as.data.frame() %>% dplyr::filter(symbols %in% c('Gzmb', 'Ifng', 'Casp1', 'Abin1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc', 'Prf1') & legend %in% c('p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj < 0.05 and L2FC < -log2(1.5)')), pch = 21, size = 4, colour = 'red') +
        scale_color_manual(values=colPal) +
        annotate("rect", xmin = -Inf, xmax = -log2(1.5), ymin = ylim.a/1.06, ymax = Inf, fill = col_WT) +
        annotate("rect", xmin = log2(1.5), xmax =Inf, ymin = ylim.a/1.06, ymax = Inf, fill = col_GTKO) +
        annotate("text", x = -log2(1.5) - diff.x.sig/20, y = ylim.a, hjust = 1, label = "Upregulated in WT", 
                 col = 'white', fontface = 2, size = 3) + 
        annotate("text", x = log2(1.5) + diff.x.sig/20, y = ylim.a, hjust = 0, label = "Upregulated in GTKO", 
                 col = 'white', fontface = 2, size = 3) +
        geom_vline(xintercept = -log2(1.5), linetype="dashed", color = col_WT, size = 0.5) +
        geom_vline(xintercept = log2(1.5), linetype="dashed", color = col_GTKO, size = 0.5) +
        geom_hline(yintercept = -log10(0.05), linetype="dashed", color = "black", size = 0.5) +
        xlim(-xlim.a, xlim.a) + ylim(0, ylim.a) + geom_text_repel(ylim = c(0, ylim.a/1.1), seed = 120, max.overlaps = 30000) + 
        ggtitle("Volcano plot of comparison between GTKO and WT mice \n (inhibited only)") + 
        theme_classic() + 
        theme(
          axis.line.x = element_line(colour = "grey50"),
          axis.line.y = element_line(colour = "grey50"),
          plot.title = element_text(face="bold"))

print(volplot1_inh)
```

Write as tiff.

```{r}
tiff('Figures/exp2_Volcano_plot_WT_vs_GTKO_inh.tiff', units = 'in', width = 6.5, height = 6.5, res = 150)
print(volplot1_inh)
dev.off()
```

We make blank version.

```{r}
tmp.tab <- res_exp2_inh %>% as.data.frame %>% 
  dplyr::filter(!is.na(padj) & !is.na(log2FoldChange)) %>%
  dplyr::mutate(log = -log10(padj), 
                log = ifelse(log == Inf, max(log[log != Inf])*1.02, log),
                legend = case_when(abs(log2FoldChange) <= log2(1.5) & padj >= .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) <= log2(1.5) & padj < .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) > log2(1.5) & padj >= .05  ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   log2FoldChange > log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC > log2(1.5)',
                                   log2FoldChange < log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC < -log2(1.5)'))
  
xlim.a <- max(abs(min(tmp.tab$log2FoldChange) - 0.1), 
              abs(max(tmp.tab$log2FoldChange) + 0.1))
ylim.a <- max(tmp.tab$log) * 1.08 
diff.x.sig <- xlim.a - 1
colPal <- c(col_WT, col_GTKO, 'gray50')
names(colPal) <- c('p-adj < 0.05 and L2FC < -log2(1.5)', 'p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj >= 0.05 or |L2FC| <= log2(1.5)')
tmp.tab$symbols <- ifelse(rownames(tmp.tab) %in% c('Gzmb', 'Ifng', 'Casp1', 'Tnip1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc'), rownames(tmp.tab), '')
tmp.tab$symbols[tmp.tab$symbols == 'Tnip1'] <- 'Abin1'
volplot1_inh_blank <- ggplot(tmp.tab, aes(log2FoldChange, log)) + 
        geom_point(aes(colour=legend), size = 2) + 
        xlab('') +
        ylab('') +
        geom_point(data = tmp.tab %>% as.data.frame() %>% dplyr::filter(symbols %in% c('Gzmb', 'Ifng', 'Casp1', 'Abin1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc') & legend %in% c('p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj < 0.05 and L2FC < -log2(1.5)')), pch = 21, size = 5, colour = 'red') +
        scale_color_manual(values=colPal) +
        annotate("rect", xmin = -Inf, xmax = -log2(1.5), ymin = ylim.a/1.06, ymax = Inf, fill = col_WT) +
        annotate("rect", xmin = log2(1.5), xmax =Inf, ymin = ylim.a/1.06, ymax = Inf, fill = col_GTKO) +
        geom_vline(xintercept = -log2(1.5), linetype="dashed", color = col_WT, size = 0.5) +
        geom_vline(xintercept = log2(1.5), linetype="dashed", color = col_GTKO, size = 0.5) +
        geom_hline(yintercept = -log10(0.05), linetype="dashed", color = "black", size = 0.5) +
        xlim(-xlim.a, xlim.a) + ylim(0, ylim.a) + 
        theme_classic() +
        ggtitle('') +
        theme(
          axis.line.x = element_line(colour = "grey50"),
          axis.line.y = element_line(colour = "grey50"),
          axis.text.x=element_blank(),
          axis.text.y=element_blank(),
          plot.title = element_text(face="bold"),
          legend.position='none'
          )

print(volplot1_inh_blank)
```

Write as tiff.

```{r}
tiff('Figures/exp2_Volcano_plot_WT_vs_GTKO_inh_blank.tiff', units = 'in', width = 4.5, height = 6.5, res = 150)
print(volplot1_inh_blank)
dev.off()
```

Prepare and generate count table with normalized counts GTKO vs. WT.

```{r}
norm.counts.inh2 <- counts(dds_exp2_inh, normalized=TRUE)
contrast.tab.inh2 <- res_exp2_inh[,c('log2FoldChange', 'pvalue', 'padj')]
colnames(contrast.tab.inh2) <- paste0(colnames(contrast.tab.inh2), '.GTKO.vs.WT') 
norm.counts.inh2 <- cbind(norm.counts.inh2, contrast.tab.inh2)
write.csv(norm.counts.inh2, 'Tables/Table_exp2_counts_contrast_GTKO_vs_WT_only_inhibited.csv')
```

# Expression using WT samples only

Now we create object using only WT samples. We split data instead of using just contrast to keep the methodology same through two experiments.

```{r}
dds_exp2_WT <- DESeqDataSet(expcounts[,expcounts$Knockout == 'WT'], design = ~ Experiment + Activation)
rowData(dds_exp2_WT)$EnsemblID <- rownames(expcounts)
dds_exp2_WT <- dds_exp2_WT[rowSums(counts(dds_exp2_WT)) > 10, ] # remove <= 10 counts = noise
dds_exp2_WT <- reannotate(dds_exp2_WT)
```

Run DESeq itself.

```{r}
dds_exp2_WT <- DESeq(dds_exp2_WT)
```

R-log normalize.

```{r}
rlog_exp2_WT <- rlog(dds_exp2_WT, blind = FALSE)
```

Generate result for desired contrast for WT data (lfc-shrunk).

```{r}
res_exp2_WT <- lfcShrink(dds = dds_exp2_WT, contrast = c('Activation', 'Act', 'Act_and_p38i'), type = 'ashr')
```

Now we plot volcano plot.

```{r}
tmp.tab <- res_exp2_WT %>% as.data.frame %>% 
  dplyr::filter(!is.na(padj) & !is.na(log2FoldChange)) %>%
  dplyr::mutate(log = -log10(padj), 
                log = ifelse(log == Inf, max(log[log != Inf])*1.02, log),
                legend = case_when(abs(log2FoldChange) <= log2(1.5) & padj >= .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) <= log2(1.5) & padj < .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) > log2(1.5) & padj >= .05  ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   log2FoldChange > log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC > log2(1.5)',
                                   log2FoldChange < log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC < -log2(1.5)'))
  
xlim.a <- max(abs(min(tmp.tab$log2FoldChange) - 0.1), 
              abs(max(tmp.tab$log2FoldChange) + 0.1))
ylim.a <- max(tmp.tab$log) * 1.08 
diff.x.sig <- xlim.a - 1
colPal <- c(col_Inh_WT, col_Act_WT, 'gray50')
names(colPal) <- c('p-adj < 0.05 and L2FC < -log2(1.5)', 'p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj >= 0.05 or |L2FC| <= log2(1.5)')
tmp.tab$symbols <- ifelse(rownames(tmp.tab) %in% c('Gzmb', 'Ifng', 'Casp1', 'Tnip1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc', 'Prf1'), rownames(tmp.tab), '')
tmp.tab$symbols[tmp.tab$symbols == 'Tnip1'] <- 'Abin1'
volplot3 <- ggplot(tmp.tab, aes(log2FoldChange, log, label= symbols)) + 
        geom_point(aes(colour=legend)) + xlab('Log2 Fold Change') + ylab('-log10(p.adj)') +
        geom_point(data = tmp.tab %>% as.data.frame() %>% dplyr::filter(symbols %in% c('Gzmb', 'Ifng', 'Casp1', 'Abin1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc', 'Prf1') & legend %in% c('p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj < 0.05 and L2FC < -log2(1.5)')), pch = 21, size = 4, colour = 'red') +
        scale_color_manual(values=colPal) +
        annotate("rect", xmin = -Inf, xmax = -log2(1.5), ymin = ylim.a/1.06, ymax = Inf, fill = col_Inh_WT) +
        annotate("rect", xmin = log2(1.5), xmax =Inf, ymin = ylim.a/1.06, ymax = Inf, fill = col_Act_WT) +
        annotate("text", x = -log2(1.5) - diff.x.sig/20, y = ylim.a, hjust = 1, label = "Upregulated in Act + p38i", 
                 col = 'black', fontface = 2, size = 3) + 
        annotate("text", x = log2(1.5) + diff.x.sig/20, y = ylim.a, hjust = 0, label = "Upregulated in Act", 
                 col = 'black', fontface = 2, size = 3) +
        geom_vline(xintercept = -log2(1.5), linetype="dashed", color = col_Inh_WT, size = 0.5) +
        geom_vline(xintercept = log2(1.5), linetype="dashed", color = col_Act_WT, size = 0.5) +
        geom_hline(yintercept = -log10(0.05), linetype="dashed", color = "black", size = 0.5) +
        xlim(-xlim.a, xlim.a) + ylim(0, ylim.a) + geom_text_repel(ylim = c(0, ylim.a/1.1), seed = 120, max.overlaps = 30000) + 
        ggtitle("Volcano plot of comparison between Activated and Inhibited cells \n from WT mice") + 
        theme_classic() + 
        theme(
          axis.line.x = element_line(colour = "grey50"),
          axis.line.y = element_line(colour = "grey50"),
          plot.title = element_text(face="bold"))

print(volplot3)
```

Write as tiff.

```{r}
tiff('Figures/exp2_Volcano_plot_Inh_vs_Act_WT.tiff', units = 'in', width = 6.5, height = 6.5, res = 150)
print(volplot3)
dev.off()
```

Make zoom of central part to see where some of genes of interest are located. This will be not used in publication.

```{r}
xlim.a <- 1
ylim.a <- max(tmp.tab$log) * 1.08 
diff.x.sig <- xlim.a - 1
colPal <- c(col_Inh_WT, col_Act_WT, 'gray50')
names(colPal) <- c('p-adj < 0.05 and L2FC < -log2(1.5)', 'p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj >= 0.05 or |L2FC| <= log2(1.5)')
tmp.tab$symbols <- ifelse(rownames(tmp.tab) %in% c('Gzmb', 'Ifng', 'Casp1', 'Tnip1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc', 'Prf1'), rownames(tmp.tab), '')
tmp.tab$symbols[tmp.tab$symbols == 'Tnip1'] <- 'Abin1'
volplot3_zoom <- ggplot(tmp.tab, aes(log2FoldChange, log, label= symbols)) + 
        geom_point(aes(colour=legend)) + xlab('Log2 Fold Change') + ylab('-log10(p.adj)') +
        geom_point(data = tmp.tab %>% as.data.frame() %>% dplyr::filter(symbols %in% c('Gzmb', 'Ifng', 'Casp1', 'Abin1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc', 'Prf1') & legend %in% c('p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj < 0.05 and L2FC < -log2(1.5)')), pch = 21, size = 4, colour = 'red') +
        scale_color_manual(values=colPal) +
        annotate("rect", xmin = -Inf, xmax = -log2(1.5), ymin = ylim.a/1.06, ymax = Inf, fill = col_Inh_WT) +
        annotate("rect", xmin = log2(1.5), xmax =Inf, ymin = ylim.a/1.06, ymax = Inf, fill = col_Act_WT) +
        geom_vline(xintercept = -log2(1.5), linetype="dashed", color = col_Inh_WT, size = 0.5) +
        geom_vline(xintercept = log2(1.5), linetype="dashed", color = col_Act_WT, size = 0.5) +
        geom_hline(yintercept = -log10(0.05), linetype="dashed", color = "black", size = 0.5) +
        xlim(-xlim.a, xlim.a) + ylim(-40, ylim.a) + geom_text_repel(ylim = c(0, ylim.a/1.1), seed = 120, max.overlaps = 30000) + 
        ggtitle("Volcano plot of comparison between Activated and Inhibited cells \n from WT mice") + 
        theme_classic() + 
        theme(
          axis.line.x = element_line(colour = "grey50"),
          axis.line.y = element_line(colour = "grey50"),
          plot.title = element_text(face="bold"))

print(volplot3_zoom)
```

Write as tiff.

```{r}
tiff('Figures/exp2_Volcano_plot_Inh_vs_Act_WT_zoom.tiff', units = 'in', width = 6.5, height = 6.5, res = 150)
print(volplot3_zoom)
dev.off()
```

We make blank version.

```{r}
xlim.a <- max(abs(min(tmp.tab$log2FoldChange) - 0.1), 
              abs(max(tmp.tab$log2FoldChange) + 0.1))
ylim.a <- max(tmp.tab$log) * 1.08 
diff.x.sig <- xlim.a - 1
colPal <- c(col_Inh_WT, col_Act_WT, 'gray50')
names(colPal) <- c('p-adj < 0.05 and L2FC < -log2(1.5)', 'p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj >= 0.05 or |L2FC| <= log2(1.5)')
tmp.tab$symbols <- ifelse(rownames(tmp.tab) %in% c('Gzmb', 'Ifng', 'Casp1', 'Tnip1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc', 'Prf1'), rownames(tmp.tab), '')
tmp.tab$symbols[tmp.tab$symbols == 'Tnip1'] <- 'Abin1'
volplot3_blank <- ggplot(tmp.tab, aes(log2FoldChange, log)) + 
        geom_point(aes(colour=legend), size = 2) + 
        xlab('') +
        ylab('') +
        geom_point(data = tmp.tab %>% as.data.frame() %>% dplyr::filter(symbols %in% c('Gzmb', 'Ifng', 'Casp1', 'Abin1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc', 'Prf1') & legend %in% c('p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj < 0.05 and L2FC < -log2(1.5)')), pch = 21, size = 5, colour = 'red') +
        scale_color_manual(values=colPal) +
        annotate("rect", xmin = -Inf, xmax = -log2(1.5), ymin = ylim.a/1.06, ymax = Inf, fill = col_Inh_WT) +
        annotate("rect", xmin = log2(1.5), xmax =Inf, ymin = ylim.a/1.06, ymax = Inf, fill = col_Act_WT) +
        geom_vline(xintercept = -log2(1.5), linetype="dashed", color = col_Inh_WT, size = 0.5) +
        geom_vline(xintercept = log2(1.5), linetype="dashed", color = col_Act_WT, size = 0.5) +
        geom_hline(yintercept = -log10(0.05), linetype="dashed", color = "black", size = 0.5) +
        xlim(-xlim.a, xlim.a) + ylim(0, ylim.a) + 
        theme_classic() +
        ggtitle('') +
        theme(
          axis.line.x = element_line(colour = "grey50"),
          axis.line.y = element_line(colour = "grey50"),
          axis.text.x=element_blank(),
          axis.text.y=element_blank(),
          plot.title = element_text(face="bold"),
          legend.position='none'
          )

print(volplot3_blank)
```

Write as tiff.

```{r}
tiff('Figures/exp2_Volcano_plot_Inh_vs_Act_WT_blank.tiff', units = 'in', width = 4.5, height = 6.5, res = 150)
print(volplot3_blank)
dev.off()
```

Prepare and generate count table with normalized counts (Activated vs. Inhibited for WT mice).

```{r}
norm.counts.wt2 <- counts(dds_exp2_WT, normalized=TRUE)
contrast.tab.wt2 <- res_exp2_WT[,c('log2FoldChange', 'pvalue', 'padj')]
colnames(contrast.tab.wt2) <- paste0(colnames(contrast.tab.wt2), '.Act.vs.Inh') 
norm.counts.wt2 <- cbind(norm.counts.wt2, contrast.tab.wt2)
write.csv(norm.counts.wt2, 'Tables/Table_exp2_counts_contrast_Act_vs_Inh_only_WT.csv')
```

# Expression using GTKO samples only

Now we create object using only GTKO samples. Reasoning for split instead of contrast is the same as above.

```{r}
dds_exp2_GTKO <- DESeqDataSet(expcounts[,expcounts$Knockout == 'GTKO'], design = ~ Experiment + Activation)
rowData(dds_exp2_GTKO)$EnsemblID <- rownames(expcounts)
dds_exp2_GTKO <- dds_exp2_GTKO[rowSums(counts(dds_exp2_GTKO)) > 10, ] # remove <= 10 counts = noise
dds_exp2_GTKO <- reannotate(dds_exp2_GTKO)
```

Run DESeq itself.

```{r}
dds_exp2_GTKO <- DESeq(dds_exp2_GTKO)
```

R-log normalize.

```{r}
rlog_exp2_GTKO <- rlog(dds_exp2_GTKO, blind = FALSE)
```

Generate result for desired contrast (lfc-shrunk).

```{r}
res_exp2_GTKO <- lfcShrink(dds = dds_exp2_GTKO, contrast = c('Activation', 'Act', 'Act_and_p38i'), type = 'ashr')
```

Now we plot volcano plot.

```{r}
tmp.tab <- res_exp2_GTKO %>% as.data.frame %>% 
  dplyr::filter(!is.na(padj) & !is.na(log2FoldChange)) %>%
  dplyr::mutate(log = -log10(padj), 
                log = ifelse(log == Inf, max(log[log != Inf])*1.02, log),
                legend = case_when(abs(log2FoldChange) <= log2(1.5) & padj >= .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) <= log2(1.5) & padj < .05 ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   abs(log2FoldChange) > log2(1.5) & padj >= .05  ~ 'p-adj >= 0.05 or |L2FC| <= log2(1.5)',
                                   log2FoldChange > log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC > log2(1.5)',
                                   log2FoldChange < log2(1.5) & padj < .05 ~ 'p-adj < 0.05 and L2FC < -log2(1.5)'))
  
xlim.a <- max(abs(min(tmp.tab$log2FoldChange) - 0.1), 
              abs(max(tmp.tab$log2FoldChange) + 0.1))
ylim.a <- max(tmp.tab$log) * 1.08 
diff.x.sig <- xlim.a - 1
colPal <- c(col_Inh_GTKO, col_Act_GTKO, 'gray50')
names(colPal) <- c('p-adj < 0.05 and L2FC < -log2(1.5)', 'p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj >= 0.05 or |L2FC| <= log2(1.5)')
tmp.tab$symbols <- ifelse(rownames(tmp.tab) %in% c('Gzmb', 'Ifng', 'Casp1', 'Tnip1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc', 'Prf1'), rownames(tmp.tab), '')
tmp.tab$symbols[tmp.tab$symbols == 'Tnip1'] <- 'Abin1'
volplot4 <- ggplot(tmp.tab, aes(log2FoldChange, log, label= symbols)) + 
        geom_point(aes(colour=legend)) + xlab('Log2 Fold Change') + ylab('-log10(p.adj)') +
        geom_point(data = tmp.tab %>% as.data.frame() %>% dplyr::filter(symbols %in% c('Gzmb', 'Ifng', 'Casp1', 'Abin1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc', 'Prf1') & legend %in% c('p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj < 0.05 and L2FC < -log2(1.5)')), pch = 21, size = 4, colour = 'red') +
        scale_color_manual(values=colPal) +
        annotate("rect", xmin = -Inf, xmax = -log2(1.5), ymin = ylim.a/1.06, ymax = Inf, fill = col_Inh_GTKO) +
        annotate("rect", xmin = log2(1.5), xmax =Inf, ymin = ylim.a/1.06, ymax = Inf, fill = col_Act_GTKO) +
        annotate("text", x = -log2(1.5) - diff.x.sig/20, y = ylim.a, hjust = 1, label = "Upregulated in Act + p38i", 
                 col = 'black', fontface = 2, size = 3) + 
        annotate("text", x = log2(1.5) + diff.x.sig/20, y = ylim.a, hjust = 0, label = "Upregulated in Act", 
                 col = 'black', fontface = 2, size = 3) +
        geom_vline(xintercept = -log2(1.5), linetype="dashed", color = col_Inh_GTKO, size = 0.5) +
        geom_vline(xintercept = log2(1.5), linetype="dashed", color = col_Act_GTKO, size = 0.5) +
        geom_hline(yintercept = -log10(0.05), linetype="dashed", color = "black", size = 0.5) +
        xlim(-xlim.a, xlim.a) + ylim(0, ylim.a) + geom_text_repel(ylim = c(0, ylim.a/1.1), seed = 120, max.overlaps = 30000) + 
        ggtitle("Volcano plot of comparison between Activated and Inhibited cells \n from GTKO mice") + 
        theme_classic() + 
        theme(
          axis.line.x = element_line(colour = "grey50"),
          axis.line.y = element_line(colour = "grey50"),
          plot.title = element_text(face="bold"))

print(volplot4)
```

Write as tiff.

```{r}
tiff('Figures/exp2_Volcano_plot_Inh_vs_Act_GTKO.tiff', units = 'in', width = 6.5, height = 6.5, res = 150)
print(volplot4)
dev.off()
```

Make zoom.

```{r}
xlim.a <- 1
ylim.a <- max(tmp.tab$log) * 1.08 
diff.x.sig <- xlim.a - 1
colPal <- c(col_Inh_GTKO, col_Act_GTKO, 'gray50')
names(colPal) <- c('p-adj < 0.05 and L2FC < -log2(1.5)', 'p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj >= 0.05 or |L2FC| <= log2(1.5)')
tmp.tab$symbols <- ifelse(rownames(tmp.tab) %in% c('Gzmb', 'Ifng', 'Casp1', 'Tnip1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc', 'Prf1'), rownames(tmp.tab), '')
tmp.tab$symbols[tmp.tab$symbols == 'Tnip1'] <- 'Abin1'
volplot4_zoom <- ggplot(tmp.tab, aes(log2FoldChange, log, label= symbols)) + 
        geom_point(aes(colour=legend)) + xlab('Log2 Fold Change') + ylab('-log10(p.adj)') +
        geom_point(data = tmp.tab %>% as.data.frame() %>% dplyr::filter(symbols %in% c('Gzmb', 'Ifng', 'Casp1', 'Abin1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc', 'Prf1') & legend %in% c('p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj < 0.05 and L2FC < -log2(1.5)')), pch = 21, size = 4, colour = 'red') +
        scale_color_manual(values=colPal) +
        annotate("rect", xmin = -Inf, xmax = -log2(1.5), ymin = ylim.a/1.06, ymax = Inf, fill = col_Inh_GTKO) +
        annotate("rect", xmin = log2(1.5), xmax =Inf, ymin = ylim.a/1.06, ymax = Inf, fill = col_Act_GTKO) +
        geom_vline(xintercept = -log2(1.5), linetype="dashed", color = col_Inh_GTKO, size = 0.5) +
        geom_vline(xintercept = log2(1.5), linetype="dashed", color = col_Act_GTKO, size = 0.5) +
        geom_hline(yintercept = -log10(0.05), linetype="dashed", color = "black", size = 0.5) +
        xlim(-xlim.a, xlim.a) + ylim(-40, ylim.a) + geom_text_repel(ylim = c(0, ylim.a/1.1), seed = 120, max.overlaps = 30000) + 
        ggtitle("Volcano plot of comparison between Activated and Inhibited cells \n from GTKO mice") + 
        theme_classic() + 
        theme(
          axis.line.x = element_line(colour = "grey50"),
          axis.line.y = element_line(colour = "grey50"),
          plot.title = element_text(face="bold"))

print(volplot4_zoom)
```

Write as tiff.

```{r}
tiff('Figures/exp2_Volcano_plot_Inh_vs_Act_GTKO_zoom.tiff', units = 'in', width = 6.5, height = 6.5, res = 150)
print(volplot4_zoom)
dev.off()
```

We make blank version.

```{r}
xlim.a <- max(abs(min(tmp.tab$log2FoldChange) - 0.1), 
              abs(max(tmp.tab$log2FoldChange) + 0.1))
ylim.a <- max(tmp.tab$log) * 1.08 
diff.x.sig <- xlim.a - 1
colPal <- c(col_Inh_GTKO, col_Act_GTKO, 'gray50')
names(colPal) <- c('p-adj < 0.05 and L2FC < -log2(1.5)', 'p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj >= 0.05 or |L2FC| <= log2(1.5)')
tmp.tab$symbols <- ifelse(rownames(tmp.tab) %in% c('Gzmb', 'Ifng', 'Casp1', 'Tnip1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc', 'Prf1'), rownames(tmp.tab), '')
tmp.tab$symbols[tmp.tab$symbols == 'Tnip1'] <- 'Abin1'
volplot4_blank <- ggplot(tmp.tab, aes(log2FoldChange, log)) + 
        geom_point(aes(colour=legend), size = 2) + 
        xlab('') +
        ylab('') +
        geom_point(data = tmp.tab %>% as.data.frame() %>% dplyr::filter(symbols %in% c('Gzmb', 'Ifng', 'Casp1', 'Abin1', 'Casp4', 'Serpina3g', 'Serpina3f', 'Il17f', 'Il17a', 'Rorc', 'Prf1') & legend %in% c('p-adj < 0.05 and L2FC > log2(1.5)', 'p-adj < 0.05 and L2FC < -log2(1.5)')), pch = 21, size = 5, colour = 'red') +
        scale_color_manual(values=colPal) +
        annotate("rect", xmin = -Inf, xmax = -log2(1.5), ymin = ylim.a/1.06, ymax = Inf, fill = col_Inh_GTKO) +
        annotate("rect", xmin = log2(1.5), xmax =Inf, ymin = ylim.a/1.06, ymax = Inf, fill = col_Act_GTKO) +
        geom_vline(xintercept = -log2(1.5), linetype="dashed", color = col_Inh_GTKO, size = 0.5) +
        geom_vline(xintercept = log2(1.5), linetype="dashed", color = col_Act_GTKO, size = 0.5) +
        geom_hline(yintercept = -log10(0.05), linetype="dashed", color = "black", size = 0.5) +
        xlim(-xlim.a, xlim.a) + ylim(0, ylim.a) + 
        theme_classic() +
        ggtitle('') +
        theme(
          axis.line.x = element_line(colour = "grey50"),
          axis.line.y = element_line(colour = "grey50"),
          axis.text.x=element_blank(),
          axis.text.y=element_blank(),
          plot.title = element_text(face="bold"),
          legend.position='none'
          )

print(volplot4_blank)
```

Write as tiff.

```{r}
tiff('Figures/exp2_Volcano_plot_Inh_vs_Act_GTKO_blank.tiff', units = 'in', width = 4.5, height = 6.5, res = 150)
print(volplot4_blank)
dev.off()
```

Prepare and generate count table with normalized counts (Activated vs. Inhibited, GTKO mice only).

```{r}
norm.counts.gtko2 <- counts(dds_exp2_GTKO, normalized=TRUE)
contrast.tab.gtko2 <- res_exp2_GTKO[,c('log2FoldChange', 'pvalue', 'padj')]
colnames(contrast.tab.gtko2) <- paste0(colnames(contrast.tab.gtko2), '.Act.vs.Inh') 
norm.counts.gtko2 <- cbind(norm.counts.gtko2, contrast.tab.gtko2)
write.csv(norm.counts.gtko2, 'Tables/Table_exp2_counts_contrast_Act_vs_Inh_only_GTKO.csv')
```

# Saving

Finally write out some of data as .rds objects for future use.

```{r}
res_exp2 <- res_exp2 %>% as.data.frame()
res_exp2$EnsemblID <- rowData(dds_exp2)$EnsemblID

res_exp2_act_vs_inh <- res_exp2_act_vs_inh %>% as.data.frame()
res_exp2_act_vs_inh$EnsemblID <- rowData(dds_exp2)$EnsemblID

res_exp2_inh <- res_exp2_inh %>% as.data.frame()
res_exp2_inh$EnsemblID <- rowData(dds_exp2_inh)$EnsemblID

res_exp2_act <- res_exp2_act %>% as.data.frame()
res_exp2_act$EnsemblID <- rowData(dds_exp2_act)$EnsemblID

saveRDS(rlog_exp2, 'Rds_data/exp2_rlog.rds')
saveRDS(rlog_exp2_WT, 'Rds_data/exp2_rlog_WT.rds')
saveRDS(rlog_exp2_GTKO, 'Rds_data/exp2_rlog_GTKO.rds')

saveRDS(res_exp2, 'Rds_data/exp2_res_KO_vs_WT.rds')
saveRDS(res_exp2_act_vs_inh, 'Rds_data/exp2_res_Act_vs_Inh.rds')
saveRDS(res_exp2_act, 'Rds_data/exp2_res_KO_vs_WT_act_only.rds')
saveRDS(res_exp2_inh, 'Rds_data/exp2_res_KO_vs_WT_inh_only.rds')

saveRDS(dds_exp2, 'Rds_data/exp2_DESeq_all.rds')
saveRDS(dds_exp2_act, 'Rds_data/exp2_DESeq_act.rds')
saveRDS(dds_exp2_WT, 'Rds_data/exp2_DESeq_WT.rds')
saveRDS(dds_exp2_GTKO, 'Rds_data/exp2_DESeq_GTKO.rds')
```

# Heatmap

Make heatmap of desired genes.

```{r}
vis_genes <- c('Ifng', 'Gzmb', 'Prf1', 'Casp1', 'Casp4', 'Serpina3f', 'Serpina3g', 'Tnfrsf4', 'Eomes', 'Ccl1', 'Ccl9', 'Il17a', 'Il17f', 'Il23r', 'Rorc')
```

Make heatmap of genes of interest.

```{r}
colfunc <- colorRampPalette(c(col_WT, "gray90", col_GTKO))
vis_genes_tab <- limma::removeBatchEffect(assay(rlog_exp2)[vis_genes,], colData(rlog_exp2)$Experiment)
#vis_genes_tab <- vis_genes_tab[,c(7:12,1:6)]

tiff('Figures/exp2_heatmap.tiff', units = 'in', width = 6.5, height = 7.5, res = 150)

pheatmap(vis_genes_tab, border = 'NA', scale = 'row', color=colfunc(64), cluster_rows = F,
         annotation_col = as.data.frame(colData(dds_exp2)[c('Activation', 'Knockout')]), cluster_cols = F,
         main='Heatmap selected genes expressed\n genes between GTKO and WT, Activated and Inhibited mice', cex = 1,
         annotation_colors = list(Knockout = c(GTKO = col_GTKO, WT = col_WT), Activation = c(Act = col_Act, Act_and_p38i = col_Inh)))

dev.off()
```

```{r}
colfunc <- colorRampPalette(c(col_WT, "gray90", col_GTKO))
vis_genes_tab <- limma::removeBatchEffect(assay(rlog_exp2)[vis_genes,], colData(rlog_exp2)$Experiment)
#vis_genes_tab <- vis_genes_tab[,c(7:12,1:6)]

tiff('Figures/exp2_heatmap_blank.tiff', units = 'in', width = 4.5, height = 5.9, res = 150)

pheatmap(vis_genes_tab, border = 'NA', scale = 'row', color=colfunc(64), cluster_rows = F,
         annotation_col = as.data.frame(colData(dds_exp2)[c('Activation','Knockout')]), cluster_cols = F,
         main='', cex = 1, annotation_names_col = F, annotation_legend = F, legend = F,
         labels_row = rep('', nrow(vis_genes_tab)),
         labels_col = rep('', ncol(vis_genes_tab)),
         annotation_colors = list(Knockout = c(GTKO = col_GTKO, WT = col_WT), Activation = c(Act = col_Act, Act_and_p38i = col_Inh)))

dev.off()
```
